// @nuxtbase/auth-ui-vue/lib/auth-ui-vue.js
import { defineComponent as N, useAttrs as ve, computed as R, openBlock as b, createBlock as L, resolveDynamicComponent as be, mergeProps as ye, unref as _, withCtx as S, renderSlot as pe, createElementBlock as B, normalizeClass as Cn, normalizeStyle as Sn, inject as $n, ref as z, watch as ut, withModifiers as q, createVNode as I, createElementVNode as k, createTextVNode as T, toDisplayString as x, createCommentVNode as j, createStaticVNode as Ne, Fragment as Fe, renderList as kn, provide as Kt, onMounted as Jt, h as Ln } from "vue";
var G = {
  SIGN_IN: "sign_in",
  SIGN_UP: "sign_up",
  FORGOTTEN_PASSWORD: "forgotten_password",
  MAGIC_LINK: "magic_link",
  UPDATE_PASSWORD: "update_password",
  VERIFY_OTP: "verify_otp"
}, Bn = "supabase-auth-ui", Tn = {
  // interfaces
  ROOT: "root",
  SIGN_IN: G.SIGN_IN,
  SIGN_UP: G.SIGN_UP,
  FORGOTTEN_PASSWORD: G.FORGOTTEN_PASSWORD,
  MAGIC_LINK: G.MAGIC_LINK,
  UPDATE_PASSWORD: G.UPDATE_PASSWORD,
  // ui
  anchor: "ui-anchor",
  button: "ui-button",
  container: "ui-container",
  divider: "ui-divider",
  input: "ui-input",
  label: "ui-label",
  loader: "ui-loader",
  message: "ui-message"
};
function we(t, a, r) {
  var n, s;
  const l = [], h = Tn[t];
  return l.push(
    r != null && r.prependedClassName ? (r == null ? void 0 : r.prependedClassName) + "_" + h : Bn + "_" + h
  ), (n = r == null ? void 0 : r.className) != null && n[t] && l.push((s = r == null ? void 0 : r.className) == null ? void 0 : s[t]), ((r == null ? void 0 : r.extend) === void 0 || (r == null ? void 0 : r.extend) === !0) && l.push(a), l;
}
function ft(t, a) {
  let r;
  if (t && a && typeof t == "object" && typeof a == "object") {
    if (Array.isArray(a))
      for (r = 0; r < a.length; r++)
        t[r] = ft(t[r], a[r]);
    else
      for (r in a)
        t[r] = ft(t[r], a[r]);
    return t;
  }
  return a;
}
function ht(t, ...a) {
  let r = a.length;
  for (let n = 0; n < r; n++)
    t = ft(t, a[n]);
  return t;
}
function xn(t, a) {
  return t.replace(
    /{{(\w+)}}/g,
    (r, n) => a.hasOwnProperty(n) ? a[n] : r
  );
}
var Mn = {
  sign_up: {
    email_label: "Email address",
    password_label: "Create a Password",
    email_input_placeholder: "Your email address",
    password_input_placeholder: "Your password",
    button_label: "Sign up",
    loading_button_label: "Signing up ...",
    social_provider_text: "Sign in with {{provider}}",
    link_text: "Don't have an account? Sign up",
    confirmation_text: "Check your email for the confirmation link"
  },
  sign_in: {
    email_label: "Email address",
    password_label: "Your Password",
    email_input_placeholder: "Your email address",
    password_input_placeholder: "Your password",
    button_label: "Sign in",
    loading_button_label: "Signing in ...",
    social_provider_text: "Sign in with {{provider}}",
    link_text: "Already have an account? Sign in"
  },
  magic_link: {
    email_input_label: "Email address",
    email_input_placeholder: "Your email address",
    button_label: "Send Magic Link",
    loading_button_label: "Sending Magic Link ...",
    link_text: "Send a magic link email",
    confirmation_text: "Check your email for the magic link"
  },
  forgotten_password: {
    email_label: "Email address",
    password_label: "Your Password",
    email_input_placeholder: "Your email address",
    button_label: "Send reset password instructions",
    loading_button_label: "Sending reset instructions ...",
    link_text: "Forgot your password?",
    confirmation_text: "Check your email for the password reset link"
  },
  update_password: {
    password_label: "New password",
    password_input_placeholder: "Your new password",
    button_label: "Update password",
    loading_button_label: "Updating password ...",
    confirmation_text: "Your password has been updated"
  },
  verify_otp: {
    email_input_label: "Email address",
    email_input_placeholder: "Your email address",
    phone_input_label: "Phone number",
    phone_input_placeholder: "Your phone number",
    token_input_label: "Token",
    token_input_placeholder: "Your Otp token",
    button_label: "Verify token",
    loading_button_label: "Signing in ..."
  }
}, Dt, A = "colors", U = "sizes", y = "space", In = { gap: y, gridGap: y, columnGap: y, gridColumnGap: y, rowGap: y, gridRowGap: y, inset: y, insetBlock: y, insetBlockEnd: y, insetBlockStart: y, insetInline: y, insetInlineEnd: y, insetInlineStart: y, margin: y, marginTop: y, marginRight: y, marginBottom: y, marginLeft: y, marginBlock: y, marginBlockEnd: y, marginBlockStart: y, marginInline: y, marginInlineEnd: y, marginInlineStart: y, padding: y, paddingTop: y, paddingRight: y, paddingBottom: y, paddingLeft: y, paddingBlock: y, paddingBlockEnd: y, paddingBlockStart: y, paddingInline: y, paddingInlineEnd: y, paddingInlineStart: y, top: y, right: y, bottom: y, left: y, scrollMargin: y, scrollMarginTop: y, scrollMarginRight: y, scrollMarginBottom: y, scrollMarginLeft: y, scrollMarginX: y, scrollMarginY: y, scrollMarginBlock: y, scrollMarginBlockEnd: y, scrollMarginBlockStart: y, scrollMarginInline: y, scrollMarginInlineEnd: y, scrollMarginInlineStart: y, scrollPadding: y, scrollPaddingTop: y, scrollPaddingRight: y, scrollPaddingBottom: y, scrollPaddingLeft: y, scrollPaddingX: y, scrollPaddingY: y, scrollPaddingBlock: y, scrollPaddingBlockEnd: y, scrollPaddingBlockStart: y, scrollPaddingInline: y, scrollPaddingInlineEnd: y, scrollPaddingInlineStart: y, fontSize: "fontSizes", background: A, backgroundColor: A, backgroundImage: A, borderImage: A, border: A, borderBlock: A, borderBlockEnd: A, borderBlockStart: A, borderBottom: A, borderBottomColor: A, borderColor: A, borderInline: A, borderInlineEnd: A, borderInlineStart: A, borderLeft: A, borderLeftColor: A, borderRight: A, borderRightColor: A, borderTop: A, borderTopColor: A, caretColor: A, color: A, columnRuleColor: A, fill: A, outline: A, outlineColor: A, stroke: A, textDecorationColor: A, fontFamily: "fonts", fontWeight: "fontWeights", lineHeight: "lineHeights", letterSpacing: "letterSpacings", blockSize: U, minBlockSize: U, maxBlockSize: U, inlineSize: U, minInlineSize: U, maxInlineSize: U, width: U, minWidth: U, maxWidth: U, height: U, minHeight: U, maxHeight: U, flexBasis: U, gridTemplateColumns: U, gridTemplateRows: U, borderWidth: "borderWidths", borderTopWidth: "borderWidths", borderRightWidth: "borderWidths", borderBottomWidth: "borderWidths", borderLeftWidth: "borderWidths", borderStyle: "borderStyles", borderTopStyle: "borderStyles", borderRightStyle: "borderStyles", borderBottomStyle: "borderStyles", borderLeftStyle: "borderStyles", borderRadius: "radii", borderTopLeftRadius: "radii", borderTopRightRadius: "radii", borderBottomRightRadius: "radii", borderBottomLeftRadius: "radii", boxShadow: "shadows", textShadow: "shadows", transition: "transitions", zIndex: "zIndices" }, zn = (t, a) => typeof a == "function" ? { "()": Function.prototype.toString.call(a) } : a, We = () => {
  const t = /* @__PURE__ */ Object.create(null);
  return (a, r, ...n) => {
    const s = ((l) => JSON.stringify(l, zn))(a);
    return s in t ? t[s] : t[s] = r(a, ...n);
  };
}, Je = Symbol.for("sxs.internal"), _t = (t, a) => Object.defineProperties(t, Object.getOwnPropertyDescriptors(a)), Ht = (t) => {
  for (const a in t)
    return !0;
  return !1;
}, { hasOwnProperty: An } = Object.prototype, gt = (t) => t.includes("-") ? t : t.replace(/[A-Z]/g, (a) => "-" + a.toLowerCase()), Pn = /\s+(?![^()]*\))/, xe = (t) => (a) => t(...typeof a == "string" ? String(a).split(Pn) : [a]), Ut = { appearance: (t) => ({ WebkitAppearance: t, appearance: t }), backfaceVisibility: (t) => ({ WebkitBackfaceVisibility: t, backfaceVisibility: t }), backdropFilter: (t) => ({ WebkitBackdropFilter: t, backdropFilter: t }), backgroundClip: (t) => ({ WebkitBackgroundClip: t, backgroundClip: t }), boxDecorationBreak: (t) => ({ WebkitBoxDecorationBreak: t, boxDecorationBreak: t }), clipPath: (t) => ({ WebkitClipPath: t, clipPath: t }), content: (t) => ({ content: t.includes('"') || t.includes("'") || /^([A-Za-z]+\([^]*|[^]*-quote|inherit|initial|none|normal|revert|unset)$/.test(t) ? t : `"${t}"` }), hyphens: (t) => ({ WebkitHyphens: t, hyphens: t }), maskImage: (t) => ({ WebkitMaskImage: t, maskImage: t }), maskSize: (t) => ({ WebkitMaskSize: t, maskSize: t }), tabSize: (t) => ({ MozTabSize: t, tabSize: t }), textSizeAdjust: (t) => ({ WebkitTextSizeAdjust: t, textSizeAdjust: t }), userSelect: (t) => ({ WebkitUserSelect: t, userSelect: t }), marginBlock: xe((t, a) => ({ marginBlockStart: t, marginBlockEnd: a || t })), marginInline: xe((t, a) => ({ marginInlineStart: t, marginInlineEnd: a || t })), maxSize: xe((t, a) => ({ maxBlockSize: t, maxInlineSize: a || t })), minSize: xe((t, a) => ({ minBlockSize: t, minInlineSize: a || t })), paddingBlock: xe((t, a) => ({ paddingBlockStart: t, paddingBlockEnd: a || t })), paddingInline: xe((t, a) => ({ paddingInlineStart: t, paddingInlineEnd: a || t })) }, dt = /([\d.]+)([^]*)/, En = (t, a) => t.length ? t.reduce((r, n) => (r.push(...a.map((s) => s.includes("&") ? s.replace(/&/g, /[ +>|~]/.test(n) && /&.*&/.test(s) ? `:is(${n})` : n) : n + " " + s)), r), []) : a, jn = (t, a) => t in Vn && typeof a == "string" ? a.replace(/^((?:[^]*[^\w-])?)(fit-content|stretch)((?:[^\w-][^]*)?)$/, (r, n, s, l) => n + (s === "stretch" ? `-moz-available${l};${gt(t)}:${n}-webkit-fill-available` : `-moz-fit-content${l};${gt(t)}:${n}fit-content`) + l) : String(a), Vn = { blockSize: 1, height: 1, inlineSize: 1, maxBlockSize: 1, maxHeight: 1, maxInlineSize: 1, maxWidth: 1, minBlockSize: 1, minHeight: 1, minInlineSize: 1, minWidth: 1, width: 1 }, oe = (t) => t ? t + "-" : "", Xt = (t, a, r) => t.replace(/([+-])?((?:\d+(?:\.\d*)?|\.\d+)(?:[Ee][+-]?\d+)?)?(\$|--)([$\w-]+)/g, (n, s, l, h, u) => h == "$" == !!l ? n : (s || h == "--" ? "calc(" : "") + "var(--" + (h === "$" ? oe(a) + (u.includes("$") ? "" : oe(r)) + u.replace(/\$/g, "-") : u) + ")" + (s || h == "--" ? "*" + (s || "") + (l || "1") + ")" : "")), Rn = /\s*,\s*(?![^()]*\))/, On = Object.prototype.toString, Me = (t, a, r, n, s) => {
  let l, h, u;
  const o = (g, f, m) => {
    let c, p;
    const v = ($) => {
      for (c in $) {
        const M = c.charCodeAt(0) === 64, $e = M && Array.isArray($[c]) ? $[c] : [$[c]];
        for (p of $e) {
          const Z = /[A-Z]/.test(w = c) ? w : w.replace(/-[^]/g, (V) => V[1].toUpperCase()), ie = typeof p == "object" && p && p.toString === On && (!n.utils[Z] || !f.length);
          if (Z in n.utils && !ie) {
            const V = n.utils[Z];
            if (V !== h) {
              h = V, v(V(p)), h = null;
              continue;
            }
          } else if (Z in Ut) {
            const V = Ut[Z];
            if (V !== u) {
              u = V, v(V(p)), u = null;
              continue;
            }
          }
          if (M && (P = c.slice(1) in n.media ? "@media " + n.media[c.slice(1)] : c, c = P.replace(/\(\s*([\w-]+)\s*(=|<|<=|>|>=)\s*([\w-]+)\s*(?:(<|<=|>|>=)\s*([\w-]+)\s*)?\)/g, (V, W, K, Q, D, J) => {
            const te = dt.test(W), de = 0.0625 * (te ? -1 : 1), [ue, Ae] = te ? [Q, W] : [W, Q];
            return "(" + (K[0] === "=" ? "" : K[0] === ">" === te ? "max-" : "min-") + ue + ":" + (K[0] !== "=" && K.length === 1 ? Ae.replace(dt, (Qe, Pe, Ee) => Number(Pe) + de * (K === ">" ? 1 : -1) + Ee) : Ae) + (D ? ") and (" + (D[0] === ">" ? "min-" : "max-") + ue + ":" + (D.length === 1 ? J.replace(dt, (Qe, Pe, Ee) => Number(Pe) + de * (D === ">" ? -1 : 1) + Ee) : J) : "") + ")";
          })), ie) {
            const V = M ? m.concat(c) : [...m], W = M ? [...f] : En(f, c.split(Rn));
            l !== void 0 && s(Gt(...l)), l = void 0, o(p, W, V);
          } else
            l === void 0 && (l = [[], f, m]), c = M || c.charCodeAt(0) !== 36 ? c : `--${oe(n.prefix)}${c.slice(1).replace(/\$/g, "-")}`, p = ie ? p : typeof p == "number" ? p && Z in Fn ? String(p) + "px" : String(p) : Xt(jn(Z, p ?? ""), n.prefix, n.themeMap[Z]), l[0].push(`${M ? `${c} ` : `${gt(c)}:`}${p}`);
        }
      }
      var P, w;
    };
    v(g), l !== void 0 && s(Gt(...l)), l = void 0;
  };
  o(t, a, r);
}, Gt = (t, a, r) => `${r.map((n) => `${n}{`).join("")}${a.length ? `${a.join(",")}{` : ""}${t.join(";")}${a.length ? "}" : ""}${Array(r.length ? r.length + 1 : 0).join("}")}`, Fn = { animationDelay: 1, animationDuration: 1, backgroundSize: 1, blockSize: 1, border: 1, borderBlock: 1, borderBlockEnd: 1, borderBlockEndWidth: 1, borderBlockStart: 1, borderBlockStartWidth: 1, borderBlockWidth: 1, borderBottom: 1, borderBottomLeftRadius: 1, borderBottomRightRadius: 1, borderBottomWidth: 1, borderEndEndRadius: 1, borderEndStartRadius: 1, borderInlineEnd: 1, borderInlineEndWidth: 1, borderInlineStart: 1, borderInlineStartWidth: 1, borderInlineWidth: 1, borderLeft: 1, borderLeftWidth: 1, borderRadius: 1, borderRight: 1, borderRightWidth: 1, borderSpacing: 1, borderStartEndRadius: 1, borderStartStartRadius: 1, borderTop: 1, borderTopLeftRadius: 1, borderTopRightRadius: 1, borderTopWidth: 1, borderWidth: 1, bottom: 1, columnGap: 1, columnRule: 1, columnRuleWidth: 1, columnWidth: 1, containIntrinsicSize: 1, flexBasis: 1, fontSize: 1, gap: 1, gridAutoColumns: 1, gridAutoRows: 1, gridTemplateColumns: 1, gridTemplateRows: 1, height: 1, inlineSize: 1, inset: 1, insetBlock: 1, insetBlockEnd: 1, insetBlockStart: 1, insetInline: 1, insetInlineEnd: 1, insetInlineStart: 1, left: 1, letterSpacing: 1, margin: 1, marginBlock: 1, marginBlockEnd: 1, marginBlockStart: 1, marginBottom: 1, marginInline: 1, marginInlineEnd: 1, marginInlineStart: 1, marginLeft: 1, marginRight: 1, marginTop: 1, maxBlockSize: 1, maxHeight: 1, maxInlineSize: 1, maxWidth: 1, minBlockSize: 1, minHeight: 1, minInlineSize: 1, minWidth: 1, offsetDistance: 1, offsetRotate: 1, outline: 1, outlineOffset: 1, outlineWidth: 1, overflowClipMargin: 1, padding: 1, paddingBlock: 1, paddingBlockEnd: 1, paddingBlockStart: 1, paddingBottom: 1, paddingInline: 1, paddingInlineEnd: 1, paddingInlineStart: 1, paddingLeft: 1, paddingRight: 1, paddingTop: 1, perspective: 1, right: 1, rowGap: 1, scrollMargin: 1, scrollMarginBlock: 1, scrollMarginBlockEnd: 1, scrollMarginBlockStart: 1, scrollMarginBottom: 1, scrollMarginInline: 1, scrollMarginInlineEnd: 1, scrollMarginInlineStart: 1, scrollMarginLeft: 1, scrollMarginRight: 1, scrollMarginTop: 1, scrollPadding: 1, scrollPaddingBlock: 1, scrollPaddingBlockEnd: 1, scrollPaddingBlockStart: 1, scrollPaddingBottom: 1, scrollPaddingInline: 1, scrollPaddingInlineEnd: 1, scrollPaddingInlineStart: 1, scrollPaddingLeft: 1, scrollPaddingRight: 1, scrollPaddingTop: 1, shapeMargin: 1, textDecoration: 1, textDecorationThickness: 1, textIndent: 1, textUnderlineOffset: 1, top: 1, transitionDelay: 1, transitionDuration: 1, verticalAlign: 1, width: 1, wordSpacing: 1 }, Zt = (t) => String.fromCharCode(t + (t > 25 ? 39 : 97)), _e = (t) => ((a) => {
  let r, n = "";
  for (r = Math.abs(a); r > 52; r = r / 52 | 0)
    n = Zt(r % 52) + n;
  return Zt(r % 52) + n;
})(((a, r) => {
  let n = r.length;
  for (; n; )
    a = 33 * a ^ r.charCodeAt(--n);
  return a;
})(5381, JSON.stringify(t)) >>> 0), Oe = ["themed", "global", "styled", "onevar", "resonevar", "allvar", "inline"], Nn = (t) => {
  if (t.href && !t.href.startsWith(location.origin))
    return !1;
  try {
    return !!t.cssRules;
  } catch {
    return !1;
  }
}, Wn = (t) => {
  let a;
  const r = () => {
    const { cssRules: s } = a.sheet;
    return [].map.call(s, (l, h) => {
      const { cssText: u } = l;
      let o = "";
      if (u.startsWith("--sxs"))
        return "";
      if (s[h - 1] && (o = s[h - 1].cssText).startsWith("--sxs")) {
        if (!l.cssRules.length)
          return "";
        for (const g in a.rules)
          if (a.rules[g].group === l)
            return `--sxs{--sxs:${[...a.rules[g].cache].join(" ")}}${u}`;
        return l.cssRules.length ? `${o}${u}` : "";
      }
      return u;
    }).join("");
  }, n = () => {
    if (a) {
      const { rules: u, sheet: o } = a;
      if (!o.deleteRule) {
        for (; Object(Object(o.cssRules)[0]).type === 3; )
          o.cssRules.splice(0, 1);
        o.cssRules = [];
      }
      for (const g in u)
        delete u[g];
    }
    const s = Object(t).styleSheets || [];
    for (const u of s)
      if (Nn(u)) {
        for (let o = 0, g = u.cssRules; g[o]; ++o) {
          const f = Object(g[o]);
          if (f.type !== 1)
            continue;
          const m = Object(g[o + 1]);
          if (m.type !== 4)
            continue;
          ++o;
          const { cssText: c } = f;
          if (!c.startsWith("--sxs"))
            continue;
          const p = c.slice(14, -3).trim().split(/\s+/), v = Oe[p[0]];
          v && (a || (a = { sheet: u, reset: n, rules: {}, toString: r }), a.rules[v] = { group: m, index: o, cache: new Set(p) });
        }
        if (a)
          break;
      }
    if (!a) {
      const u = (o, g) => ({ type: g, cssRules: [], insertRule(f, m) {
        this.cssRules.splice(m, 0, u(f, { import: 3, undefined: 1 }[(f.toLowerCase().match(/^@([a-z]+)/) || [])[1]] || 4));
      }, get cssText() {
        return o === "@media{}" ? `@media{${[].map.call(this.cssRules, (f) => f.cssText).join("")}}` : o;
      } });
      a = { sheet: t ? (t.head || t).appendChild(document.createElement("style")).sheet : u("", "text/css"), rules: {}, reset: n, toString: r };
    }
    const { sheet: l, rules: h } = a;
    for (let u = Oe.length - 1; u >= 0; --u) {
      const o = Oe[u];
      if (!h[o]) {
        const g = Oe[u + 1], f = h[g] ? h[g].index : l.cssRules.length;
        l.insertRule("@media{}", f), l.insertRule(`--sxs{--sxs:${u}}`, f), h[o] = { group: l.cssRules[f + 1], index: f, cache: /* @__PURE__ */ new Set([u]) };
      }
      Dn(h[o]);
    }
  };
  return n(), a;
}, Dn = (t) => {
  const a = t.group;
  let r = a.cssRules.length;
  t.apply = (n) => {
    try {
      a.insertRule(n, r), ++r;
    } catch {
    }
  };
}, Re = Symbol(), Hn = We(), Un = (t, a) => Hn(t, () => (...r) => {
  let n = { type: null, composers: /* @__PURE__ */ new Set() };
  for (const s of r)
    if (s != null)
      if (s[Je]) {
        n.type == null && (n.type = s[Je].type);
        for (const l of s[Je].composers)
          n.composers.add(l);
      } else
        s.constructor !== Object || s.$$typeof ? n.type == null && (n.type = s) : n.composers.add(Gn(s, t));
  return n.type == null && (n.type = "span"), n.composers.size || n.composers.add(["PJLV", {}, [], [], {}, []]), Zn(t, n, a);
}), Gn = ({ variants: t, compoundVariants: a, defaultVariants: r, ...n }, s) => {
  const l = `${oe(s.prefix)}c-${_e(n)}`, h = [], u = [], o = /* @__PURE__ */ Object.create(null), g = [];
  for (const c in r)
    o[c] = String(r[c]);
  if (typeof t == "object" && t)
    for (const c in t) {
      f = o, m = c, An.call(f, m) || (o[c] = "undefined");
      const p = t[c];
      for (const v in p) {
        const $ = { [c]: String(v) };
        String(v) === "undefined" && g.push(c);
        const P = p[v], w = [$, P, !Ht(P)];
        h.push(w);
      }
    }
  var f, m;
  if (typeof a == "object" && a)
    for (const c of a) {
      let { css: p, ...v } = c;
      p = typeof p == "object" && p || {};
      for (const P in v)
        v[P] = String(v[P]);
      const $ = [v, p, !Ht(p)];
      u.push($);
    }
  return [l, n, h, u, o, g];
}, Zn = (t, a, r) => {
  const [n, s, l, h] = Yn(a.composers), u = typeof a.type == "function" || a.type.$$typeof ? ((m) => {
    function c() {
      for (let p = 0; p < c[Re].length; p++) {
        const [v, $] = c[Re][p];
        m.rules[v].apply($);
      }
      return c[Re] = [], null;
    }
    return c[Re] = [], c.rules = {}, Oe.forEach((p) => c.rules[p] = { apply: (v) => c[Re].push([p, v]) }), c;
  })(r) : null, o = (u || r).rules, g = `.${n}${s.length > 1 ? `:where(.${s.slice(1).join(".")})` : ""}`, f = (m) => {
    m = typeof m == "object" && m || qn;
    const { css: c, ...p } = m, v = {};
    for (const w in l)
      if (delete p[w], w in m) {
        let M = m[w];
        typeof M == "object" && M ? v[w] = { "@initial": l[w], ...M } : (M = String(M), v[w] = M !== "undefined" || h.has(w) ? M : l[w]);
      } else
        v[w] = l[w];
    const $ = /* @__PURE__ */ new Set([...s]);
    for (const [w, M, $e, Z] of a.composers) {
      r.rules.styled.cache.has(w) || (r.rules.styled.cache.add(w), Me(M, [`.${w}`], [], t, (W) => {
        o.styled.apply(W);
      }));
      const ie = Yt($e, v, t.media), V = Yt(Z, v, t.media, !0);
      for (const W of ie)
        if (W !== void 0)
          for (const [K, Q, D] of W) {
            const J = `${w}-${_e(Q)}-${K}`;
            $.add(J);
            const te = (D ? r.rules.resonevar : r.rules.onevar).cache, de = D ? o.resonevar : o.onevar;
            te.has(J) || (te.add(J), Me(Q, [`.${J}`], [], t, (ue) => {
              de.apply(ue);
            }));
          }
      for (const W of V)
        if (W !== void 0)
          for (const [K, Q] of W) {
            const D = `${w}-${_e(Q)}-${K}`;
            $.add(D), r.rules.allvar.cache.has(D) || (r.rules.allvar.cache.add(D), Me(Q, [`.${D}`], [], t, (J) => {
              o.allvar.apply(J);
            }));
          }
    }
    if (typeof c == "object" && c) {
      const w = `${n}-i${_e(c)}-css`;
      $.add(w), r.rules.inline.cache.has(w) || (r.rules.inline.cache.add(w), Me(c, [`.${w}`], [], t, (M) => {
        o.inline.apply(M);
      }));
    }
    for (const w of String(m.className || "").trim().split(/\s+/))
      w && $.add(w);
    const P = p.className = [...$].join(" ");
    return { type: a.type, className: P, selector: g, props: p, toString: () => P, deferredInjector: u };
  };
  return _t(f, { className: n, selector: g, [Je]: a, toString: () => (r.rules.styled.cache.has(n) || f(), n) });
}, Yn = (t) => {
  let a = "";
  const r = [], n = {}, s = [];
  for (const [l, , , , h, u] of t) {
    a === "" && (a = l), r.push(l), s.push(...u);
    for (const o in h) {
      const g = h[o];
      (n[o] === void 0 || g !== "undefined" || u.includes(g)) && (n[o] = g);
    }
  }
  return [a, r, n, new Set(s)];
}, Yt = (t, a, r, n) => {
  const s = [];
  e:
    for (let [l, h, u] of t) {
      if (u)
        continue;
      let o, g = 0, f = !1;
      for (o in l) {
        const m = l[o];
        let c = a[o];
        if (c !== m) {
          if (typeof c != "object" || !c)
            continue e;
          {
            let p, v, $ = 0;
            for (const P in c) {
              if (m === String(c[P])) {
                if (P !== "@initial") {
                  const w = P.slice(1);
                  (v = v || []).push(w in r ? r[w] : P.replace(/^@media ?/, "")), f = !0;
                }
                g += $, p = !0;
              }
              ++$;
            }
            if (v && v.length && (h = { ["@media " + v.join(", ")]: h }), !p)
              continue e;
          }
        }
      }
      (s[g] = s[g] || []).push([n ? "cv" : `${o}-${l[o]}`, h, f]);
    }
  return s;
}, qn = {}, Kn = We(), Jn = (t, a) => Kn(t, () => (...r) => {
  const n = () => {
    for (let s of r) {
      s = typeof s == "object" && s || {};
      let l = _e(s);
      if (!a.rules.global.cache.has(l)) {
        if (a.rules.global.cache.add(l), "@import" in s) {
          let h = [].indexOf.call(a.sheet.cssRules, a.rules.themed.group) - 1;
          for (let u of [].concat(s["@import"]))
            u = u.includes('"') || u.includes("'") ? u : `"${u}"`, a.sheet.insertRule(`@import ${u};`, h++);
          delete s["@import"];
        }
        Me(s, [], [], t, (h) => {
          a.rules.global.apply(h);
        });
      }
    }
    return "";
  };
  return _t(n, { toString: n });
}), Xn = We(), Qn = (t, a) => Xn(t, () => (r) => {
  const n = `${oe(t.prefix)}k-${_e(r)}`, s = () => {
    if (!a.rules.global.cache.has(n)) {
      a.rules.global.cache.add(n);
      const l = [];
      Me(r, [], [], t, (u) => l.push(u));
      const h = `@keyframes ${n}{${l.join("")}}`;
      a.rules.global.apply(h);
    }
    return n;
  };
  return _t(s, { get name() {
    return s();
  }, toString: s });
}), er = class {
  constructor(t, a, r, n) {
    this.token = t == null ? "" : String(t), this.value = a == null ? "" : String(a), this.scale = r == null ? "" : String(r), this.prefix = n == null ? "" : String(n);
  }
  get computedValue() {
    return "var(" + this.variable + ")";
  }
  get variable() {
    return "--" + oe(this.prefix) + oe(this.scale) + this.token;
  }
  toString() {
    return this.computedValue;
  }
}, tr = We(), ar = (t, a) => tr(t, () => (r, n) => {
  n = typeof r == "object" && r || Object(n);
  const s = `.${r = (r = typeof r == "string" ? r : "") || `${oe(t.prefix)}t-${_e(n)}`}`, l = {}, h = [];
  for (const o in n) {
    l[o] = {};
    for (const g in n[o]) {
      const f = `--${oe(t.prefix)}${o}-${g}`, m = Xt(String(n[o][g]), t.prefix, o);
      l[o][g] = new er(g, m, o, t.prefix), h.push(`${f}:${m}`);
    }
  }
  const u = () => {
    if (h.length && !a.rules.themed.cache.has(r)) {
      a.rules.themed.cache.add(r);
      const o = `${n === t.theme ? ":root," : ""}.${r}{${h.join(";")}}`;
      a.rules.themed.apply(o);
    }
    return r;
  };
  return { ...l, get className() {
    return u();
  }, selector: s, toString: u };
}), nr = We(), mt = (t) => {
  let a = !1;
  const r = nr(t, (n) => {
    a = !0;
    const s = "prefix" in (n = typeof n == "object" && n || {}) ? String(n.prefix) : "", l = typeof n.media == "object" && n.media || {}, h = typeof n.root == "object" ? n.root || null : globalThis.document || null, u = typeof n.theme == "object" && n.theme || {}, o = { prefix: s, media: l, theme: u, themeMap: typeof n.themeMap == "object" && n.themeMap || { ...In }, utils: typeof n.utils == "object" && n.utils || {} }, g = Wn(h), f = { css: Un(o, g), globalCss: Jn(o, g), keyframes: Qn(o, g), createTheme: ar(o, g), reset() {
      g.reset(), f.theme.toString();
    }, theme: {}, sheet: g, config: o, prefix: s, getCssText: g.toString, toString: g.toString };
    return String(f.theme = f.createTheme(u)), f;
  });
  return a || r.reset(), r;
}, Qt = () => Dt || (Dt = mt()), rr = (...t) => Qt().createTheme(...t), Ce = (...t) => Qt().css(...t), qe = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function or(t) {
  return t && t.__esModule && Object.prototype.hasOwnProperty.call(t, "default") ? t.default : t;
}
var Xe = { exports: {} };
Xe.exports;
(function(t, a) {
  var r = 200, n = "__lodash_hash_undefined__", s = 9007199254740991, l = "[object Arguments]", h = "[object Array]", u = "[object Boolean]", o = "[object Date]", g = "[object Error]", f = "[object Function]", m = "[object GeneratorFunction]", c = "[object Map]", p = "[object Number]", v = "[object Object]", $ = "[object Promise]", P = "[object RegExp]", w = "[object Set]", M = "[object String]", $e = "[object Symbol]", Z = "[object WeakMap]", ie = "[object ArrayBuffer]", V = "[object DataView]", W = "[object Float32Array]", K = "[object Float64Array]", Q = "[object Int8Array]", D = "[object Int16Array]", J = "[object Int32Array]", te = "[object Uint8Array]", de = "[object Uint8ClampedArray]", ue = "[object Uint16Array]", Ae = "[object Uint32Array]", Qe = /[\\^$.*+?()[\]{}|]/g, Pe = /\w*$/, Ee = /^\[object .+?Constructor\]$/, ea = /^(?:0|[1-9]\d*)$/, E = {};
  E[l] = E[h] = E[ie] = E[V] = E[u] = E[o] = E[W] = E[K] = E[Q] = E[D] = E[J] = E[c] = E[p] = E[v] = E[P] = E[w] = E[M] = E[$e] = E[te] = E[de] = E[ue] = E[Ae] = !0, E[g] = E[f] = E[Z] = !1;
  var ta = typeof qe == "object" && qe && qe.Object === Object && qe, aa = typeof self == "object" && self && self.Object === Object && self, ae = ta || aa || Function("return this")(), vt = a && !a.nodeType && a, bt = vt && !0 && t && !t.nodeType && t, na = bt && bt.exports === vt;
  function ra(e, i) {
    return e.set(i[0], i[1]), e;
  }
  function oa(e, i) {
    return e.add(i), e;
  }
  function ia(e, i) {
    for (var d = -1, C = e ? e.length : 0; ++d < C && i(e[d], d, e) !== !1; )
      ;
    return e;
  }
  function la(e, i) {
    for (var d = -1, C = i.length, F = e.length; ++d < C; )
      e[F + d] = i[d];
    return e;
  }
  function yt(e, i, d, C) {
    var F = -1, H = e ? e.length : 0;
    for (C && H && (d = e[++F]); ++F < H; )
      d = i(d, e[F], F, e);
    return d;
  }
  function sa(e, i) {
    for (var d = -1, C = Array(e); ++d < e; )
      C[d] = i(d);
    return C;
  }
  function ca(e, i) {
    return e == null ? void 0 : e[i];
  }
  function wt(e) {
    var i = !1;
    if (e != null && typeof e.toString != "function")
      try {
        i = !!(e + "");
      } catch {
      }
    return i;
  }
  function Ct(e) {
    var i = -1, d = Array(e.size);
    return e.forEach(function(C, F) {
      d[++i] = [F, C];
    }), d;
  }
  function et(e, i) {
    return function(d) {
      return e(i(d));
    };
  }
  function St(e) {
    var i = -1, d = Array(e.size);
    return e.forEach(function(C) {
      d[++i] = C;
    }), d;
  }
  var pa = Array.prototype, da = Function.prototype, De = Object.prototype, tt = ae["__core-js_shared__"], $t = function() {
    var e = /[^.]+$/.exec(tt && tt.keys && tt.keys.IE_PROTO || "");
    return e ? "Symbol(src)_1." + e : "";
  }(), kt = da.toString, le = De.hasOwnProperty, He = De.toString, ua = RegExp(
    "^" + kt.call(le).replace(Qe, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
  ), Lt = na ? ae.Buffer : void 0, Bt = ae.Symbol, Tt = ae.Uint8Array, fa = et(Object.getPrototypeOf, Object), ha = Object.create, ga = De.propertyIsEnumerable, ma = pa.splice, xt = Object.getOwnPropertySymbols, _a = Lt ? Lt.isBuffer : void 0, va = et(Object.keys, Object), at = Be(ae, "DataView"), je = Be(ae, "Map"), nt = Be(ae, "Promise"), rt = Be(ae, "Set"), ot = Be(ae, "WeakMap"), Ve = Be(Object, "create"), ba = ge(at), ya = ge(je), wa = ge(nt), Ca = ge(rt), Sa = ge(ot), Mt = Bt ? Bt.prototype : void 0, It = Mt ? Mt.valueOf : void 0;
  function fe(e) {
    var i = -1, d = e ? e.length : 0;
    for (this.clear(); ++i < d; ) {
      var C = e[i];
      this.set(C[0], C[1]);
    }
  }
  function $a() {
    this.__data__ = Ve ? Ve(null) : {};
  }
  function ka(e) {
    return this.has(e) && delete this.__data__[e];
  }
  function La(e) {
    var i = this.__data__;
    if (Ve) {
      var d = i[e];
      return d === n ? void 0 : d;
    }
    return le.call(i, e) ? i[e] : void 0;
  }
  function Ba(e) {
    var i = this.__data__;
    return Ve ? i[e] !== void 0 : le.call(i, e);
  }
  function Ta(e, i) {
    var d = this.__data__;
    return d[e] = Ve && i === void 0 ? n : i, this;
  }
  fe.prototype.clear = $a, fe.prototype.delete = ka, fe.prototype.get = La, fe.prototype.has = Ba, fe.prototype.set = Ta;
  function ne(e) {
    var i = -1, d = e ? e.length : 0;
    for (this.clear(); ++i < d; ) {
      var C = e[i];
      this.set(C[0], C[1]);
    }
  }
  function xa() {
    this.__data__ = [];
  }
  function Ma(e) {
    var i = this.__data__, d = Ue(i, e);
    if (d < 0)
      return !1;
    var C = i.length - 1;
    return d == C ? i.pop() : ma.call(i, d, 1), !0;
  }
  function Ia(e) {
    var i = this.__data__, d = Ue(i, e);
    return d < 0 ? void 0 : i[d][1];
  }
  function za(e) {
    return Ue(this.__data__, e) > -1;
  }
  function Aa(e, i) {
    var d = this.__data__, C = Ue(d, e);
    return C < 0 ? d.push([e, i]) : d[C][1] = i, this;
  }
  ne.prototype.clear = xa, ne.prototype.delete = Ma, ne.prototype.get = Ia, ne.prototype.has = za, ne.prototype.set = Aa;
  function ke(e) {
    var i = -1, d = e ? e.length : 0;
    for (this.clear(); ++i < d; ) {
      var C = e[i];
      this.set(C[0], C[1]);
    }
  }
  function Pa() {
    this.__data__ = {
      hash: new fe(),
      map: new (je || ne)(),
      string: new fe()
    };
  }
  function Ea(e) {
    return Ge(this, e).delete(e);
  }
  function ja(e) {
    return Ge(this, e).get(e);
  }
  function Va(e) {
    return Ge(this, e).has(e);
  }
  function Ra(e, i) {
    return Ge(this, e).set(e, i), this;
  }
  ke.prototype.clear = Pa, ke.prototype.delete = Ea, ke.prototype.get = ja, ke.prototype.has = Va, ke.prototype.set = Ra;
  function Le(e) {
    this.__data__ = new ne(e);
  }
  function Oa() {
    this.__data__ = new ne();
  }
  function Fa(e) {
    return this.__data__.delete(e);
  }
  function Na(e) {
    return this.__data__.get(e);
  }
  function Wa(e) {
    return this.__data__.has(e);
  }
  function Da(e, i) {
    var d = this.__data__;
    if (d instanceof ne) {
      var C = d.__data__;
      if (!je || C.length < r - 1)
        return C.push([e, i]), this;
      d = this.__data__ = new ke(C);
    }
    return d.set(e, i), this;
  }
  Le.prototype.clear = Oa, Le.prototype.delete = Fa, Le.prototype.get = Na, Le.prototype.has = Wa, Le.prototype.set = Da;
  function Ha(e, i) {
    var d = st(e) || gn(e) ? sa(e.length, String) : [], C = d.length, F = !!C;
    for (var H in e)
      (i || le.call(e, H)) && !(F && (H == "length" || dn(H, C))) && d.push(H);
    return d;
  }
  function zt(e, i, d) {
    var C = e[i];
    (!(le.call(e, i) && jt(C, d)) || d === void 0 && !(i in e)) && (e[i] = d);
  }
  function Ue(e, i) {
    for (var d = e.length; d--; )
      if (jt(e[d][0], i))
        return d;
    return -1;
  }
  function Ua(e, i) {
    return e && At(i, ct(i), e);
  }
  function it(e, i, d, C, F, H, X) {
    var Y;
    if (C && (Y = H ? C(e, F, H, X) : C(e)), Y !== void 0)
      return Y;
    if (!Ze(e))
      return e;
    var Ot = st(e);
    if (Ot) {
      if (Y = sn(e), !i)
        return rn(e, Y);
    } else {
      var Te = he(e), Ft = Te == f || Te == m;
      if (_n(e))
        return Ja(e, i);
      if (Te == v || Te == l || Ft && !H) {
        if (wt(e))
          return H ? e : {};
        if (Y = cn(Ft ? {} : e), !i)
          return on(e, Ua(Y, e));
      } else {
        if (!E[Te])
          return H ? e : {};
        Y = pn(e, Te, it, i);
      }
    }
    X || (X = new Le());
    var Nt = X.get(e);
    if (Nt)
      return Nt;
    if (X.set(e, Y), !Ot)
      var Wt = d ? ln(e) : ct(e);
    return ia(Wt || e, function(pt, Ye) {
      Wt && (Ye = pt, pt = e[Ye]), zt(Y, Ye, it(pt, i, d, C, Ye, e, X));
    }), Y;
  }
  function Ga(e) {
    return Ze(e) ? ha(e) : {};
  }
  function Za(e, i, d) {
    var C = i(e);
    return st(e) ? C : la(C, d(e));
  }
  function Ya(e) {
    return He.call(e);
  }
  function qa(e) {
    if (!Ze(e) || fn(e))
      return !1;
    var i = Rt(e) || wt(e) ? ua : Ee;
    return i.test(ge(e));
  }
  function Ka(e) {
    if (!Et(e))
      return va(e);
    var i = [];
    for (var d in Object(e))
      le.call(e, d) && d != "constructor" && i.push(d);
    return i;
  }
  function Ja(e, i) {
    if (i)
      return e.slice();
    var d = new e.constructor(e.length);
    return e.copy(d), d;
  }
  function lt(e) {
    var i = new e.constructor(e.byteLength);
    return new Tt(i).set(new Tt(e)), i;
  }
  function Xa(e, i) {
    var d = i ? lt(e.buffer) : e.buffer;
    return new e.constructor(d, e.byteOffset, e.byteLength);
  }
  function Qa(e, i, d) {
    var C = i ? d(Ct(e), !0) : Ct(e);
    return yt(C, ra, new e.constructor());
  }
  function en(e) {
    var i = new e.constructor(e.source, Pe.exec(e));
    return i.lastIndex = e.lastIndex, i;
  }
  function tn(e, i, d) {
    var C = i ? d(St(e), !0) : St(e);
    return yt(C, oa, new e.constructor());
  }
  function an(e) {
    return It ? Object(It.call(e)) : {};
  }
  function nn(e, i) {
    var d = i ? lt(e.buffer) : e.buffer;
    return new e.constructor(d, e.byteOffset, e.length);
  }
  function rn(e, i) {
    var d = -1, C = e.length;
    for (i || (i = Array(C)); ++d < C; )
      i[d] = e[d];
    return i;
  }
  function At(e, i, d, C) {
    d || (d = {});
    for (var F = -1, H = i.length; ++F < H; ) {
      var X = i[F], Y = C ? C(d[X], e[X], X, d, e) : void 0;
      zt(d, X, Y === void 0 ? e[X] : Y);
    }
    return d;
  }
  function on(e, i) {
    return At(e, Pt(e), i);
  }
  function ln(e) {
    return Za(e, ct, Pt);
  }
  function Ge(e, i) {
    var d = e.__data__;
    return un(i) ? d[typeof i == "string" ? "string" : "hash"] : d.map;
  }
  function Be(e, i) {
    var d = ca(e, i);
    return qa(d) ? d : void 0;
  }
  var Pt = xt ? et(xt, Object) : yn, he = Ya;
  (at && he(new at(new ArrayBuffer(1))) != V || je && he(new je()) != c || nt && he(nt.resolve()) != $ || rt && he(new rt()) != w || ot && he(new ot()) != Z) && (he = function(e) {
    var i = He.call(e), d = i == v ? e.constructor : void 0, C = d ? ge(d) : void 0;
    if (C)
      switch (C) {
        case ba:
          return V;
        case ya:
          return c;
        case wa:
          return $;
        case Ca:
          return w;
        case Sa:
          return Z;
      }
    return i;
  });
  function sn(e) {
    var i = e.length, d = e.constructor(i);
    return i && typeof e[0] == "string" && le.call(e, "index") && (d.index = e.index, d.input = e.input), d;
  }
  function cn(e) {
    return typeof e.constructor == "function" && !Et(e) ? Ga(fa(e)) : {};
  }
  function pn(e, i, d, C) {
    var F = e.constructor;
    switch (i) {
      case ie:
        return lt(e);
      case u:
      case o:
        return new F(+e);
      case V:
        return Xa(e, C);
      case W:
      case K:
      case Q:
      case D:
      case J:
      case te:
      case de:
      case ue:
      case Ae:
        return nn(e, C);
      case c:
        return Qa(e, C, d);
      case p:
      case M:
        return new F(e);
      case P:
        return en(e);
      case w:
        return tn(e, C, d);
      case $e:
        return an(e);
    }
  }
  function dn(e, i) {
    return i = i ?? s, !!i && (typeof e == "number" || ea.test(e)) && e > -1 && e % 1 == 0 && e < i;
  }
  function un(e) {
    var i = typeof e;
    return i == "string" || i == "number" || i == "symbol" || i == "boolean" ? e !== "__proto__" : e === null;
  }
  function fn(e) {
    return !!$t && $t in e;
  }
  function Et(e) {
    var i = e && e.constructor, d = typeof i == "function" && i.prototype || De;
    return e === d;
  }
  function ge(e) {
    if (e != null) {
      try {
        return kt.call(e);
      } catch {
      }
      try {
        return e + "";
      } catch {
      }
    }
    return "";
  }
  function hn(e) {
    return it(e, !0, !0);
  }
  function jt(e, i) {
    return e === i || e !== e && i !== i;
  }
  function gn(e) {
    return mn(e) && le.call(e, "callee") && (!ga.call(e, "callee") || He.call(e) == l);
  }
  var st = Array.isArray;
  function Vt(e) {
    return e != null && vn(e.length) && !Rt(e);
  }
  function mn(e) {
    return bn(e) && Vt(e);
  }
  var _n = _a || wn;
  function Rt(e) {
    var i = Ze(e) ? He.call(e) : "";
    return i == f || i == m;
  }
  function vn(e) {
    return typeof e == "number" && e > -1 && e % 1 == 0 && e <= s;
  }
  function Ze(e) {
    var i = typeof e;
    return !!e && (i == "object" || i == "function");
  }
  function bn(e) {
    return !!e && typeof e == "object";
  }
  function ct(e) {
    return Vt(e) ? Ha(e) : Ka(e);
  }
  function yn() {
    return [];
  }
  function wn() {
    return !1;
  }
  t.exports = hn;
})(Xe, Xe.exports);
var ir = Xe.exports;
const Ke = /* @__PURE__ */ or(ir), Se = Symbol("AuthView"), lr = Symbol("UserContextProvider"), sr = /* @__PURE__ */ N({
  __name: "SocialAuthContainer",
  props: {
    theme: {},
    className: {},
    appearance: {}
  },
  setup(t) {
    const a = t, r = ve(), n = R(() => {
      var s, l, h, u;
      return a.theme !== "default" ? rr(
        ht(
          (l = (s = a.appearance) == null ? void 0 : s.theme) == null ? void 0 : l[a.theme],
          ((u = (h = a.appearance) == null ? void 0 : h.variables) == null ? void 0 : u[a.theme]) ?? {}
        )
      ).className : "";
    });
    return (s, l) => (b(), L(be("div"), ye(_(r), { class: n.value }), {
      default: S(() => [
        pe(s.$slots, "default")
      ]),
      _: 3
    }, 16, ["class"]));
  }
}), me = /* @__PURE__ */ N({
  __name: "Anchor",
  props: {
    appearance: {}
  },
  setup(t) {
    const a = t, r = Ce({
      fontFamily: "$bodyFontFamily",
      fontSize: "$baseBodySize",
      marginBottom: "$anchorBottomMargin",
      color: "$anchorTextColor",
      display: "block",
      textAlign: "center",
      textDecoration: "underline",
      "&:hover": {
        color: "$anchorTextHoverColor"
      }
    }), n = ve(), s = R(() => we(
      "anchor",
      r(),
      a.appearance
    ).join(" "));
    return (l, h) => {
      var u, o;
      return b(), L(be("a"), ye(_(n), {
        class: s.value,
        style: (o = (u = l.appearance) == null ? void 0 : u.style) == null ? void 0 : o.anchor
      }), {
        default: S(() => [
          pe(l.$slots, "default")
        ]),
        _: 3
      }, 16, ["class", "style"]);
    };
  }
}), cr = ["disabled"], Ie = /* @__PURE__ */ N({
  __name: "Button",
  props: {
    appearance: {},
    icon: {},
    color: { default: "default" },
    loading: { type: Boolean, default: !1 }
  },
  setup(t) {
    const a = t, r = Ce({
      fontFamily: "$buttonFontFamily",
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      gap: "8px",
      borderRadius: "$borderRadiusButton",
      fontSize: "$baseButtonSize",
      padding: "$buttonPadding",
      cursor: "pointer",
      borderWidth: "$buttonBorderWidth",
      borderStyle: "solid",
      width: "100%",
      transitionProperty: "background-color",
      transitionTimingFunction: "cubic-bezier(0.4, 0, 0.2, 1)",
      transitionDuration: "100ms",
      "&:disabled": {
        opacity: 0.7,
        cursor: "unset"
      },
      variants: {
        color: {
          default: {
            backgroundColor: "$defaultButtonBackground",
            color: "$defaultButtonText",
            borderColor: "$defaultButtonBorder",
            "&:hover:not(:disabled)": {
              backgroundColor: "$defaultButtonBackgroundHover"
            }
          },
          primary: {
            backgroundColor: "$brand",
            color: "$brandButtonText",
            borderColor: "$brandAccent",
            "&:hover:not(:disabled)": {
              backgroundColor: "$brandAccent"
            }
          }
        }
      }
    }), n = ve(), s = R(() => we(
      "button",
      r({ color: a.color }),
      a.appearance
    ).join(" "));
    return (l, h) => {
      var u, o;
      return b(), B("button", ye(_(n), {
        class: s.value,
        style: (o = (u = l.appearance) == null ? void 0 : u.style) == null ? void 0 : o.button,
        disabled: l.loading
      }), [
        pe(l.$slots, "icon"),
        pe(l.$slots, "default")
      ], 16, cr);
    };
  }
}), re = /* @__PURE__ */ N({
  __name: "Container",
  props: {
    is: { default: "div" },
    direction: {},
    gap: {},
    appearance: {}
  },
  setup(t) {
    const a = t, r = Ce({
      display: "flex",
      gap: "4px",
      variants: {
        direction: {
          horizontal: {
            display: "grid",
            gridTemplateColumns: "repeat(auto-fit, minmax(48px, 1fr))"
          },
          vertical: {
            flexDirection: "column",
            margin: "8px 0"
          }
        },
        gap: {
          small: {
            gap: "4px"
          },
          medium: {
            gap: "8px"
          },
          large: {
            gap: "16px"
          }
        }
      }
    }), n = R(() => we(
      "container",
      r({
        direction: a.direction,
        gap: a.gap
      }),
      a.appearance
    ).join(" "));
    return (s, l) => {
      var h, u;
      return b(), L(be(s.is), {
        class: Cn(n.value),
        style: Sn((u = (h = s.appearance) == null ? void 0 : h.style) == null ? void 0 : u.container)
      }, {
        default: S(() => [
          pe(s.$slots, "default")
        ]),
        _: 3
      }, 8, ["class", "style"]);
    };
  }
}), pr = /* @__PURE__ */ N({
  __name: "Divider",
  props: {
    appearance: {}
  },
  setup(t) {
    const a = t, r = Ce({
      background: "$dividerBackground",
      display: "block",
      margin: "16px 0",
      height: "1px",
      width: "100%"
    }), n = ve(), s = R(() => we(
      "divider",
      r(),
      a.appearance
    ).join(" "));
    return (l, h) => {
      var u, o;
      return b(), L(be("div"), ye(_(n), {
        class: s.value,
        style: (o = (u = l.appearance) == null ? void 0 : u.style) == null ? void 0 : o.divider
      }), {
        default: S(() => [
          pe(l.$slots, "default")
        ]),
        _: 3
      }, 16, ["class", "style"]);
    };
  }
}), dr = ["type", "value"], se = /* @__PURE__ */ N({
  __name: "Input",
  props: {
    type: {},
    appearance: {},
    modelValue: {}
  },
  emits: ["update:modelValue"],
  setup(t) {
    const a = t, r = Ce({
      fontFamily: "$inputFontFamily",
      background: "$inputBackground",
      borderRadius: "$inputBorderRadius",
      padding: "$inputPadding",
      cursor: "text",
      borderWidth: "$inputBorderWidth",
      borderColor: "$inputBorder",
      borderStyle: "solid",
      fontSize: "$baseInputSize",
      width: "100%",
      color: "$inputText",
      boxSizing: "border-box",
      "&:hover": {
        borderColor: "$inputBorderHover",
        outline: "none"
      },
      "&:focus": {
        borderColor: "$inputBorderFocus",
        outline: "none"
      },
      "&::placeholder": {
        color: "$inputPlaceholder",
        letterSpacing: "initial"
      },
      transitionProperty: "background-color, border",
      transitionTimingFunction: "cubic-bezier(0.4, 0, 0.2, 1)",
      transitionDuration: "100ms",
      variants: {
        type: {
          default: {
            letterSpacing: "0px"
          },
          password: {
            letterSpacing: "6px"
          }
        }
      }
    }), n = ve(), s = R(() => we(
      "input",
      r({
        type: a.type === "password" ? "password" : "default"
      }),
      a.appearance
    ).join(" "));
    return (l, h) => {
      var u, o;
      return b(), B("input", ye(_(n), {
        type: l.type,
        class: s.value,
        style: (o = (u = l.appearance) == null ? void 0 : u.style) == null ? void 0 : o.input,
        value: l.modelValue,
        onInput: h[0] || (h[0] = (g) => l.$emit("update:modelValue", (g == null ? void 0 : g.target).value))
      }), null, 16, dr);
    };
  }
}), ce = /* @__PURE__ */ N({
  __name: "Label",
  props: {
    appearance: {}
  },
  setup(t) {
    const a = t, r = Ce({
      fontFamily: "$labelFontFamily",
      fontSize: "$baseLabelSize",
      marginBottom: "$labelBottomMargin",
      color: "$inputLabelText",
      display: "block"
    }), n = ve(), s = R(() => we(
      "label",
      r(),
      a.appearance
    ).join(" "));
    return (l, h) => {
      var u, o;
      return b(), L(be("label"), ye(_(n), {
        class: s.value,
        style: (o = (u = l.appearance) == null ? void 0 : u.style) == null ? void 0 : o.label
      }), {
        default: S(() => [
          pe(l.$slots, "default")
        ]),
        _: 3
      }, 16, ["class", "style"]);
    };
  }
}), ee = /* @__PURE__ */ N({
  __name: "Message",
  props: {
    color: { default: "default" },
    appearance: {}
  },
  setup(t) {
    const a = t, r = Ce({
      fontFamily: "$bodyFontFamily",
      fontSize: "$baseBodySize",
      marginBottom: "$labelBottomMargin",
      display: "block",
      textAlign: "center",
      variants: {
        color: {
          default: {
            color: "$messageText"
          },
          danger: {
            color: "$messageTextDanger"
          }
        }
      }
    }), n = ve(), s = R(() => we(
      "message",
      r({ color: a.color }),
      a.appearance
    ).join(" "));
    return (l, h) => {
      var u, o;
      return b(), L(be("span"), ye(_(n), {
        class: s.value,
        style: (o = (u = l.appearance) == null ? void 0 : u.style) == null ? void 0 : o.message
      }), {
        default: S(() => [
          pe(l.$slots, "default")
        ]),
        _: 3
      }, 16, ["class", "style"]);
    };
  }
});
function ze(t, a) {
  const r = $n(t, a);
  if (!r)
    throw new Error(`Could not resolve ${t.description}`);
  return r;
}
const ur = ["id", "onSubmit"], fr = /* @__PURE__ */ N({
  __name: "EmailAuth",
  props: {
    appearance: {},
    supabaseClient: {},
    defaultEmail: {},
    defaultPassword: {},
    redirectTo: {},
    magicLink: { type: Boolean },
    showLinks: { type: Boolean },
    i18n: {},
    additionalData: {}
  },
  setup(t) {
    const a = t, r = z(""), n = z(""), s = z(""), l = z(""), h = z(!1), { authView: u, setAuthView: o } = ze(Se), g = R(
      () => {
        var m;
        return (m = a.i18n) == null ? void 0 : m[u.value];
      }
    ), f = async (m) => {
      var c, p;
      switch (s.value = "", l.value = "", h.value = !0, u.value) {
        case "sign_in":
          const {
            // data: { user: signInUser, session: signInSession },
            error: v
          } = await a.supabaseClient.auth.signInWithPassword({
            email: r.value,
            password: n.value
          });
          v && (s.value = v.message), h.value = !1;
          break;
        case "sign_up":
          let $ = {
            emailRedirectTo: a.redirectTo
          };
          a.additionalData && ($.data = a.additionalData);
          const {
            data: { user: P, session: w },
            error: M
          } = await a.supabaseClient.auth.signUp({
            email: r.value,
            password: n.value,
            options: $
          });
          M ? s.value = M.message : P && !w && (l.value = (p = (c = a.i18n) == null ? void 0 : c.sign_up) == null ? void 0 : p.confirmation_text), h.value = !1;
      }
    };
    return ut(
      () => u.value,
      () => {
        h.value = !1;
      }
    ), (m, c) => (b(), B("form", {
      id: `auth-${_(u)}`,
      onSubmit: q(f, ["prevent"]),
      autoComplete: "on",
      style: {
        width: "100%"
      }
    }, [
      I(_(re), {
        direction: "vertical",
        gap: "large",
        appearance: m.appearance
      }, {
        default: S(() => [
          I(_(re), {
            direction: "vertical",
            gap: "large",
            appearance: m.appearance
          }, {
            default: S(() => {
              var p, v;
              return [
                k("div", null, [
                  I(_(ce), {
                    htmlFor: "email",
                    appearance: m.appearance
                  }, {
                    default: S(() => {
                      var $;
                      return [
                        T(x(($ = g.value) == null ? void 0 : $.email_label), 1)
                      ];
                    }),
                    _: 1
                  }, 8, ["appearance"]),
                  I(_(se), {
                    id: "email",
                    type: "email",
                    name: "email",
                    modelValue: r.value,
                    "onUpdate:modelValue": c[0] || (c[0] = ($) => r.value = $),
                    placeholder: (p = g.value) == null ? void 0 : p.email_input_placeholder,
                    autoComplete: "email",
                    appearance: m.appearance
                  }, null, 8, ["modelValue", "placeholder", "appearance"])
                ]),
                k("div", null, [
                  I(_(ce), {
                    htmlFor: "password",
                    appearance: m.appearance
                  }, {
                    default: S(() => {
                      var $;
                      return [
                        T(x(($ = g.value) == null ? void 0 : $.password_label), 1)
                      ];
                    }),
                    _: 1
                  }, 8, ["appearance"]),
                  I(_(se), {
                    id: "password",
                    type: "password",
                    name: "password",
                    modelValue: n.value,
                    "onUpdate:modelValue": c[1] || (c[1] = ($) => n.value = $),
                    placeholder: (v = g.value) == null ? void 0 : v.password_input_placeholder,
                    autoComplete: _(u) === "sign_in" ? "current-password" : "new-password",
                    appearance: m.appearance
                  }, null, 8, ["modelValue", "placeholder", "autoComplete", "appearance"])
                ])
              ];
            }),
            _: 1
          }, 8, ["appearance"]),
          I(_(Ie), {
            type: "submit",
            color: "primary",
            loading: h.value,
            appearance: m.appearance
          }, {
            default: S(() => {
              var p, v;
              return [
                T(x(h.value ? (p = g.value) == null ? void 0 : p.loading_button_label : (v = g.value) == null ? void 0 : v.button_label), 1)
              ];
            }),
            _: 1
          }, 8, ["loading", "appearance"]),
          m.showLinks ? (b(), L(_(re), {
            key: 0,
            direction: "vertical",
            gap: "small",
            appearance: m.appearance
          }, {
            default: S(() => [
              _(u) === _(G).SIGN_IN && m.magicLink ? (b(), L(_(me), {
                key: 0,
                href: "#auth-magic-link",
                appearance: m.appearance,
                onClick: c[2] || (c[2] = q((p) => _(o)(_(G).MAGIC_LINK), ["prevent"]))
              }, {
                default: S(() => {
                  var p, v;
                  return [
                    T(x((v = (p = m.i18n) == null ? void 0 : p.magic_link) == null ? void 0 : v.link_text), 1)
                  ];
                }),
                _: 1
              }, 8, ["appearance"])) : j("", !0),
              _(u) === _(G).SIGN_IN ? (b(), L(_(me), {
                key: 1,
                href: "#auth-forgot-password",
                appearance: m.appearance,
                onClick: c[3] || (c[3] = q((p) => _(o)(_(G).FORGOTTEN_PASSWORD), ["prevent"]))
              }, {
                default: S(() => {
                  var p, v;
                  return [
                    T(x((v = (p = m.i18n) == null ? void 0 : p.forgotten_password) == null ? void 0 : v.link_text), 1)
                  ];
                }),
                _: 1
              }, 8, ["appearance"])) : j("", !0),
              _(u) === _(G).SIGN_IN ? (b(), L(_(me), {
                key: 2,
                href: "#auth-sign-up",
                appearance: m.appearance,
                onClick: c[4] || (c[4] = q((p) => _(o)(_(G).SIGN_UP), ["prevent"]))
              }, {
                default: S(() => {
                  var p, v;
                  return [
                    T(x((v = (p = m.i18n) == null ? void 0 : p.sign_up) == null ? void 0 : v.link_text), 1)
                  ];
                }),
                _: 1
              }, 8, ["appearance"])) : (b(), L(_(me), {
                key: 3,
                href: "#auth-sign-in",
                appearance: m.appearance,
                onClick: c[5] || (c[5] = q((p) => _(o)(_(G).SIGN_IN), ["prevent"]))
              }, {
                default: S(() => {
                  var p, v;
                  return [
                    T(x((v = (p = m.i18n) == null ? void 0 : p.sign_in) == null ? void 0 : v.link_text), 1)
                  ];
                }),
                _: 1
              }, 8, ["appearance"]))
            ]),
            _: 1
          }, 8, ["appearance"])) : j("", !0)
        ]),
        _: 1
      }, 8, ["appearance"]),
      l.value ? (b(), L(_(ee), {
        key: 0,
        appearance: m.appearance
      }, {
        default: S(() => [
          T(x(l.value), 1)
        ]),
        _: 1
      }, 8, ["appearance"])) : j("", !0),
      s.value ? (b(), L(_(ee), {
        key: 1,
        color: "danger",
        appearance: m.appearance
      }, {
        default: S(() => [
          T(x(s.value), 1)
        ]),
        _: 1
      }, 8, ["appearance"])) : j("", !0)
    ], 40, ur));
  }
}), O = (t, a) => {
  const r = t.__vccOpts || t;
  for (const [n, s] of a)
    r[n] = s;
  return r;
}, hr = {}, gr = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 48 48",
  width: "21px",
  height: "21px"
}, mr = /* @__PURE__ */ k("path", {
  fill: "#FFC107",
  d: "M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
}, null, -1), _r = /* @__PURE__ */ k("path", {
  fill: "#FF3D00",
  d: "M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
}, null, -1), vr = /* @__PURE__ */ k("path", {
  fill: "#4CAF50",
  d: "M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
}, null, -1), br = /* @__PURE__ */ k("path", {
  fill: "#1976D2",
  d: "M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
}, null, -1), yr = [
  mr,
  _r,
  vr,
  br
];
function wr(t, a) {
  return b(), B("svg", gr, yr);
}
const Cr = /* @__PURE__ */ O(hr, [["render", wr]]), Sr = {}, $r = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 48 48",
  width: "21px",
  height: "21px"
}, kr = /* @__PURE__ */ k("path", {
  fill: "#039be5",
  d: "M24 5A19 19 0 1 0 24 43A19 19 0 1 0 24 5Z"
}, null, -1), Lr = /* @__PURE__ */ k("path", {
  fill: "#fff",
  d: "M26.572,29.036h4.917l0.772-4.995h-5.69v-2.73c0-2.075,0.678-3.915,2.619-3.915h3.119v-4.359c-0.548-0.074-1.707-0.236-3.897-0.236c-4.573,0-7.254,2.415-7.254,7.917v3.323h-4.701v4.995h4.701v13.729C22.089,42.905,23.032,43,24,43c0.875,0,1.729-0.08,2.572-0.194V29.036z"
}, null, -1), Br = [
  kr,
  Lr
];
function Tr(t, a) {
  return b(), B("svg", $r, Br);
}
const xr = /* @__PURE__ */ O(Sr, [["render", Tr]]), Mr = {}, Ir = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 48 48",
  width: "21px",
  height: "21px"
}, zr = /* @__PURE__ */ k("path", {
  fill: "#03A9F4",
  d: "M42,12.429c-1.323,0.586-2.746,0.977-4.247,1.162c1.526-0.906,2.7-2.351,3.251-4.058c-1.428,0.837-3.01,1.452-4.693,1.776C34.967,9.884,33.05,9,30.926,9c-4.08,0-7.387,3.278-7.387,7.32c0,0.572,0.067,1.129,0.193,1.67c-6.138-0.308-11.582-3.226-15.224-7.654c-0.64,1.082-1,2.349-1,3.686c0,2.541,1.301,4.778,3.285,6.096c-1.211-0.037-2.351-0.374-3.349-0.914c0,0.022,0,0.055,0,0.086c0,3.551,2.547,6.508,5.923,7.181c-0.617,0.169-1.269,0.263-1.941,0.263c-0.477,0-0.942-0.054-1.392-0.135c0.94,2.902,3.667,5.023,6.898,5.086c-2.528,1.96-5.712,3.134-9.174,3.134c-0.598,0-1.183-0.034-1.761-0.104C9.268,36.786,13.152,38,17.321,38c13.585,0,21.017-11.156,21.017-20.834c0-0.317-0.01-0.633-0.025-0.945C39.763,15.197,41.013,13.905,42,12.429"
}, null, -1), Ar = [
  zr
];
function Pr(t, a) {
  return b(), B("svg", Ir, Ar);
}
const Er = /* @__PURE__ */ O(Mr, [["render", Pr]]), jr = {}, Vr = {
  fill: "gray",
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 24 24",
  width: "21px",
  height: "21px"
}, Rr = /* @__PURE__ */ k("path", { d: "M 15.904297 1.078125 C 15.843359 1.06875 15.774219 1.0746094 15.699219 1.0996094 C 14.699219 1.2996094 13.600391 1.8996094 12.900391 2.5996094 C 12.300391 3.1996094 11.800781 4.1996094 11.800781 5.0996094 C 11.800781 5.2996094 11.999219 5.5 12.199219 5.5 C 13.299219 5.4 14.399609 4.7996094 15.099609 4.0996094 C 15.699609 3.2996094 16.199219 2.4 16.199219 1.5 C 16.199219 1.275 16.087109 1.10625 15.904297 1.078125 z M 16.199219 5.4003906 C 14.399219 5.4003906 13.600391 6.5 12.400391 6.5 C 11.100391 6.5 9.9003906 5.5 8.4003906 5.5 C 6.3003906 5.5 3.0996094 7.4996094 3.0996094 12.099609 C 2.9996094 16.299609 6.8 21 9 21 C 10.3 21 10.600391 20.199219 12.400391 20.199219 C 14.200391 20.199219 14.600391 21 15.900391 21 C 17.400391 21 18.500391 19.399609 19.400391 18.099609 C 19.800391 17.399609 20.100391 17.000391 20.400391 16.400391 C 20.600391 16.000391 20.4 15.600391 20 15.400391 C 17.4 14.100391 16.900781 9.9003906 19.800781 8.4003906 C 20.300781 8.1003906 20.4 7.4992188 20 7.1992188 C 18.9 6.1992187 17.299219 5.4003906 16.199219 5.4003906 z" }, null, -1), Or = [
  Rr
];
function Fr(t, a) {
  return b(), B("svg", Vr, Or);
}
const Nr = /* @__PURE__ */ O(jr, [["render", Fr]]), Wr = {}, Dr = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "21.34",
  height: "21",
  viewBox: "0 0 256 384"
}, Hr = /* @__PURE__ */ Ne('<path fill="#0ACF83" d="M64 384c35.328 0 64-28.672 64-64v-64H64c-35.328 0-64 28.672-64 64s28.672 64 64 64Z"></path><path fill="#A259FF" d="M0 192c0-35.328 28.672-64 64-64h64v128H64c-35.328 0-64-28.672-64-64Z"></path><path fill="#F24E1E" d="M0 64C0 28.672 28.672 0 64 0h64v128H64C28.672 128 0 99.328 0 64Z"></path><path fill="#FF7262" d="M128 0h64c35.328 0 64 28.672 64 64s-28.672 64-64 64h-64V0Z"></path><path fill="#1ABCFE" d="M256 192c0 35.328-28.672 64-64 64s-64-28.672-64-64s28.672-64 64-64s64 28.672 64 64Z"></path>', 5), Ur = [
  Hr
];
function Gr(t, a) {
  return b(), B("svg", Dr, Ur);
}
const Zr = /* @__PURE__ */ O(Wr, [["render", Gr]]), Yr = {}, qr = {
  fill: "gray",
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 30 30",
  width: "21px",
  height: "21px"
}, Kr = /* @__PURE__ */ k("path", { d: "M15,3C8.373,3,3,8.373,3,15c0,5.623,3.872,10.328,9.092,11.63C12.036,26.468,12,26.28,12,26.047v-2.051 c-0.487,0-1.303,0-1.508,0c-0.821,0-1.551-0.353-1.905-1.009c-0.393-0.729-0.461-1.844-1.435-2.526 c-0.289-0.227-0.069-0.486,0.264-0.451c0.615,0.174,1.125,0.596,1.605,1.222c0.478,0.627,0.703,0.769,1.596,0.769 c0.433,0,1.081-0.025,1.691-0.121c0.328-0.833,0.895-1.6,1.588-1.962c-3.996-0.411-5.903-2.399-5.903-5.098 c0-1.162,0.495-2.286,1.336-3.233C9.053,10.647,8.706,8.73,9.435,8c1.798,0,2.885,1.166,3.146,1.481C13.477,9.174,14.461,9,15.495,9 c1.036,0,2.024,0.174,2.922,0.483C18.675,9.17,19.763,8,21.565,8c0.732,0.731,0.381,2.656,0.102,3.594 c0.836,0.945,1.328,2.066,1.328,3.226c0,2.697-1.904,4.684-5.894,5.097C18.199,20.49,19,22.1,19,23.313v2.734 c0,0.104-0.023,0.179-0.035,0.268C23.641,24.676,27,20.236,27,15C27,8.373,21.627,3,15,3z" }, null, -1), Jr = [
  Kr
];
function Xr(t, a) {
  return b(), B("svg", qr, Jr);
}
const Qr = /* @__PURE__ */ O(Yr, [["render", Xr]]), eo = {}, to = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 48 48",
  width: "21px",
  height: "21px"
}, ao = /* @__PURE__ */ Ne('<path fill="#e53935" d="M24 43L16 20 32 20z"></path><path fill="#ff7043" d="M24 43L42 20 32 20z"></path><path fill="#e53935" d="M37 5L42 20 32 20z"></path><path fill="#ffa726" d="M24 43L42 20 45 28z"></path><path fill="#ff7043" d="M24 43L6 20 16 20z"></path><path fill="#e53935" d="M11 5L6 20 16 20z"></path><path fill="#ffa726" d="M24 43L6 20 3 28z"></path>', 7), no = [
  ao
];
function ro(t, a) {
  return b(), B("svg", to, no);
}
const oo = /* @__PURE__ */ O(eo, [["render", ro]]), io = {}, lo = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "21",
  height: "21",
  viewBox: "0 0 62.42 62.42"
}, so = /* @__PURE__ */ Ne('<defs><linearGradient id="New_Gradient_Swatch_1" x1="64.01" y1="30.27" x2="32.99" y2="54.48" gradientUnits="userSpaceOnUse"><stop offset="0.18" stop-color="#0052cc"></stop><stop offset="1" stop-color="#2684ff"></stop></linearGradient></defs><title>Bitbucket-blue</title><g id="Layer_2" data-name="Layer 2"><g id="Blue" transform="translate(0 -3.13)"><path d="M2,6.26A2,2,0,0,0,0,8.58L8.49,60.12a2.72,2.72,0,0,0,2.66,2.27H51.88a2,2,0,0,0,2-1.68L62.37,8.59a2,2,0,0,0-2-2.32ZM37.75,43.51h-13L21.23,25.12H40.9Z" fill="#2684ff"></path><path d="M59.67,25.12H40.9L37.75,43.51h-13L9.4,61.73a2.71,2.71,0,0,0,1.75.66H51.89a2,2,0,0,0,2-1.68Z" fill="url(#New_Gradient_Swatch_1)"></path></g></g>', 3), co = [
  so
];
function po(t, a) {
  return b(), B("svg", lo, co);
}
const uo = /* @__PURE__ */ O(io, [["render", po]]), fo = {}, ho = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 48 48",
  width: "21px",
  height: "21px"
}, go = /* @__PURE__ */ k("path", {
  fill: "#536dfe",
  d: "M39.248,10.177c-2.804-1.287-5.812-2.235-8.956-2.778c-0.057-0.01-0.114,0.016-0.144,0.068	c-0.387,0.688-0.815,1.585-1.115,2.291c-3.382-0.506-6.747-0.506-10.059,0c-0.3-0.721-0.744-1.603-1.133-2.291	c-0.03-0.051-0.087-0.077-0.144-0.068c-3.143,0.541-6.15,1.489-8.956,2.778c-0.024,0.01-0.045,0.028-0.059,0.051	c-5.704,8.522-7.267,16.835-6.5,25.044c0.003,0.04,0.026,0.079,0.057,0.103c3.763,2.764,7.409,4.442,10.987,5.554	c0.057,0.017,0.118-0.003,0.154-0.051c0.846-1.156,1.601-2.374,2.248-3.656c0.038-0.075,0.002-0.164-0.076-0.194	c-1.197-0.454-2.336-1.007-3.432-1.636c-0.087-0.051-0.094-0.175-0.014-0.234c0.231-0.173,0.461-0.353,0.682-0.534	c0.04-0.033,0.095-0.04,0.142-0.019c7.201,3.288,14.997,3.288,22.113,0c0.047-0.023,0.102-0.016,0.144,0.017	c0.22,0.182,0.451,0.363,0.683,0.536c0.08,0.059,0.075,0.183-0.012,0.234c-1.096,0.641-2.236,1.182-3.434,1.634	c-0.078,0.03-0.113,0.12-0.075,0.196c0.661,1.28,1.415,2.498,2.246,3.654c0.035,0.049,0.097,0.07,0.154,0.052	c3.595-1.112,7.241-2.79,11.004-5.554c0.033-0.024,0.054-0.061,0.057-0.101c0.917-9.491-1.537-17.735-6.505-25.044	C39.293,10.205,39.272,10.187,39.248,10.177z M16.703,30.273c-2.168,0-3.954-1.99-3.954-4.435s1.752-4.435,3.954-4.435	c2.22,0,3.989,2.008,3.954,4.435C20.658,28.282,18.906,30.273,16.703,30.273z M31.324,30.273c-2.168,0-3.954-1.99-3.954-4.435	s1.752-4.435,3.954-4.435c2.22,0,3.989,2.008,3.954,4.435C35.278,28.282,33.544,30.273,31.324,30.273z"
}, null, -1), mo = [
  go
];
function _o(t, a) {
  return b(), B("svg", ho, mo);
}
const vo = /* @__PURE__ */ O(fo, [["render", _o]]), bo = {}, yo = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 48 48",
  width: "21px",
  height: "21px"
}, wo = /* @__PURE__ */ Ne('<linearGradient id="k8yl7~hDat~FaoWq8WjN6a" x1="-1254.397" x2="-1261.911" y1="877.268" y2="899.466" gradientTransform="translate(1981.75 -1362.063) scale(1.5625)" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="#114a8b"></stop><stop offset="1" stop-color="#0669bc"></stop></linearGradient><path fill="url(#k8yl7~hDat~FaoWq8WjN6a)" d="M17.634,6h11.305L17.203,40.773c-0.247,0.733-0.934,1.226-1.708,1.226H6.697 c-0.994,0-1.8-0.806-1.8-1.8c0-0.196,0.032-0.39,0.094-0.576L15.926,7.227C16.173,6.494,16.86,6,17.634,6L17.634,6z"></path><path fill="#0078d4" d="M34.062,29.324H16.135c-0.458-0.001-0.83,0.371-0.831,0.829c0,0.231,0.095,0.451,0.264,0.608 l11.52,10.752C27.423,41.826,27.865,42,28.324,42h10.151L34.062,29.324z"></path><linearGradient id="k8yl7~hDat~FaoWq8WjN6b" x1="-1252.05" x2="-1253.788" y1="887.612" y2="888.2" gradientTransform="translate(1981.75 -1362.063) scale(1.5625)" gradientUnits="userSpaceOnUse"><stop offset="0" stop-opacity=".3"></stop><stop offset=".071" stop-opacity=".2"></stop><stop offset=".321" stop-opacity=".1"></stop><stop offset=".623" stop-opacity=".05"></stop><stop offset="1" stop-opacity="0"></stop></linearGradient><path fill="url(#k8yl7~hDat~FaoWq8WjN6b)" d="M17.634,6c-0.783-0.003-1.476,0.504-1.712,1.25L5.005,39.595 c-0.335,0.934,0.151,1.964,1.085,2.299C6.286,41.964,6.493,42,6.702,42h9.026c0.684-0.122,1.25-0.603,1.481-1.259l2.177-6.416 l7.776,7.253c0.326,0.27,0.735,0.419,1.158,0.422h10.114l-4.436-12.676l-12.931,0.003L28.98,6H17.634z"></path><linearGradient id="k8yl7~hDat~FaoWq8WjN6c" x1="-1252.952" x2="-1244.704" y1="876.6" y2="898.575" gradientTransform="translate(1981.75 -1362.063) scale(1.5625)" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="#3ccbf4"></stop><stop offset="1" stop-color="#2892df"></stop></linearGradient><path fill="url(#k8yl7~hDat~FaoWq8WjN6c)" d="M32.074,7.225C31.827,6.493,31.141,6,30.368,6h-12.6c0.772,0,1.459,0.493,1.705,1.224 l10.935,32.399c0.318,0.942-0.188,1.963-1.13,2.281C29.093,41.968,28.899,42,28.703,42h12.6c0.994,0,1.8-0.806,1.8-1.801 c0-0.196-0.032-0.39-0.095-0.575L32.074,7.225z"></path>', 7), Co = [
  wo
];
function So(t, a) {
  return b(), B("svg", yo, Co);
}
const $o = /* @__PURE__ */ O(bo, [["render", So]]), ko = {}, Lo = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "21",
  height: "21",
  viewBox: "0 0 256 256"
}, Bo = /* @__PURE__ */ k("path", {
  fill: "#FFE812",
  d: "M256 236c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20V20C0 8.954 8.954 0 20 0h216c11.046 0 20 8.954 20 20v216z"
}, null, -1), To = /* @__PURE__ */ k("path", { d: "M128 36C70.562 36 24 72.713 24 118c0 29.279 19.466 54.97 48.748 69.477-1.593 5.494-10.237 35.344-10.581 37.689 0 0-.207 1.762.934 2.434s2.483.15 2.483.15c3.272-.457 37.943-24.811 43.944-29.04 5.995.849 12.168 1.29 18.472 1.29 57.438 0 104-36.712 104-82 0-45.287-46.562-82-104-82z" }, null, -1), xo = /* @__PURE__ */ k("path", {
  fill: "#FFE812",
  d: "M70.5 146.625c-3.309 0-6-2.57-6-5.73V105.25h-9.362c-3.247 0-5.888-2.636-5.888-5.875s2.642-5.875 5.888-5.875h30.724c3.247 0 5.888 2.636 5.888 5.875s-2.642 5.875-5.888 5.875H76.5v35.645c0 3.16-2.691 5.73-6 5.73zM123.112 146.547c-2.502 0-4.416-1.016-4.993-2.65l-2.971-7.778-18.296-.001-2.973 7.783c-.575 1.631-2.488 2.646-4.99 2.646a9.155 9.155 0 0 1-3.814-.828c-1.654-.763-3.244-2.861-1.422-8.52l14.352-37.776c1.011-2.873 4.082-5.833 7.99-5.922 3.919.088 6.99 3.049 8.003 5.928l14.346 37.759c1.826 5.672.236 7.771-1.418 8.532a9.176 9.176 0 0 1-3.814.827c-.001 0 0 0 0 0zm-11.119-21.056L106 108.466l-5.993 17.025h11.986zM138 145.75c-3.171 0-5.75-2.468-5.75-5.5V99.5c0-3.309 2.748-6 6.125-6s6.125 2.691 6.125 6v35.25h12.75c3.171 0 5.75 2.468 5.75 5.5s-2.579 5.5-5.75 5.5H138zM171.334 146.547c-3.309 0-6-2.691-6-6V99.5c0-3.309 2.691-6 6-6s6 2.691 6 6v12.896l16.74-16.74c.861-.861 2.044-1.335 3.328-1.335 1.498 0 3.002.646 4.129 1.772 1.051 1.05 1.678 2.401 1.764 3.804.087 1.415-.384 2.712-1.324 3.653l-13.673 13.671 14.769 19.566a5.951 5.951 0 0 1 1.152 4.445 5.956 5.956 0 0 1-2.328 3.957 5.94 5.94 0 0 1-3.609 1.211 5.953 5.953 0 0 1-4.793-2.385l-14.071-18.644-2.082 2.082v13.091a6.01 6.01 0 0 1-6.002 6.003z"
}, null, -1), Mo = [
  Bo,
  To,
  xo
];
function Io(t, a) {
  return b(), B("svg", Lo, Mo);
}
const zo = /* @__PURE__ */ O(ko, [["render", Io]]), Ao = {}, Po = {
  width: "21",
  height: "21",
  viewBox: "0 0 512 512",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
}, Eo = /* @__PURE__ */ k("path", {
  d: "M472.136 163.959H408.584C407.401 163.959 406.218 163.327 405.666 162.3L354.651 73.6591C354.02 72.632 352.916 72 351.654 72H143.492C142.309 72 141.126 72.632 140.574 73.6591L87.5084 165.618L36.414 254.259C35.862 255.286 35.862 256.55 36.414 257.656L87.5084 346.297L140.495 438.335C141.047 439.362 142.23 440.073 143.413 439.994H351.654C352.837 439.994 354.02 439.362 354.651 438.335L405.745 349.694C406.297 348.667 407.48 347.956 408.663 348.035H472.215C474.344 348.035 476 346.297 476 344.243V167.83C475.921 165.697 474.186 163.959 472.136 163.959ZM228.728 349.694L212.721 377.345C212.485 377.74 212.091 378.135 211.696 378.372C211.223 378.609 210.75 378.767 210.198 378.767H178.422C177.318 378.767 176.293 378.214 175.82 377.187L128.431 294.787L123.779 286.65L106.748 257.498C106.511 257.103 106.353 256.629 106.432 256.076C106.432 255.602 106.59 255.049 106.827 254.654L123.937 224.949L175.899 134.886C176.451 133.938 177.476 133.306 178.501 133.306H210.198C210.75 133.306 211.302 133.464 211.854 133.701C212.248 133.938 212.643 134.254 212.879 134.728L228.886 162.537C229.359 163.485 229.28 164.67 228.728 165.539L177.397 254.654C177.16 255.049 177.081 255.523 177.081 255.918C177.081 256.392 177.239 256.787 177.397 257.182L228.728 346.218C229.438 347.403 229.359 348.667 228.728 349.694V349.694ZM388.083 257.498L371.051 286.65L366.399 294.787L319.011 377.187C318.459 378.135 317.512 378.767 316.409 378.767H284.632C284.08 378.767 283.607 378.609 283.134 378.372C282.74 378.135 282.346 377.819 282.109 377.345L266.103 349.694C265.393 348.667 265.393 347.403 266.024 346.376L317.355 257.34C317.591 256.945 317.67 256.471 317.67 256.076C317.67 255.602 317.513 255.207 317.355 254.812L266.024 165.697C265.472 164.749 265.393 163.643 265.866 162.695L281.873 134.886C282.109 134.491 282.503 134.096 282.898 133.859C283.371 133.543 283.923 133.464 284.553 133.464H316.409C317.512 133.464 318.538 134.017 319.011 135.044L370.972 225.107L388.083 254.812C388.319 255.286 388.477 255.76 388.477 256.234C388.477 256.55 388.319 257.024 388.083 257.498V257.498Z",
  fill: "#008AAA"
}, null, -1), jo = [
  Eo
];
function Vo(t, a) {
  return b(), B("svg", Po, jo);
}
const Ro = /* @__PURE__ */ O(Ao, [["render", Vo]]), Oo = {}, Fo = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 48 48",
  width: "21px",
  height: "21px"
}, No = /* @__PURE__ */ k("path", {
  fill: "#0288D1",
  d: "M42,37c0,2.762-2.238,5-5,5H11c-2.761,0-5-2.238-5-5V11c0-2.762,2.239-5,5-5h26c2.762,0,5,2.238,5,5V37z"
}, null, -1), Wo = /* @__PURE__ */ k("path", {
  fill: "#FFF",
  d: "M12 19H17V36H12zM14.485 17h-.028C12.965 17 12 15.888 12 14.499 12 13.08 12.995 12 14.514 12c1.521 0 2.458 1.08 2.486 2.499C17 15.887 16.035 17 14.485 17zM36 36h-5v-9.099c0-2.198-1.225-3.698-3.192-3.698-1.501 0-2.313 1.012-2.707 1.99C24.957 25.543 25 26.511 25 27v9h-5V19h5v2.616C25.721 20.5 26.85 19 29.738 19c3.578 0 6.261 2.25 6.261 7.274L36 36 36 36z"
}, null, -1), Do = [
  No,
  Wo
];
function Ho(t, a) {
  return b(), B("svg", Fo, Do);
}
const qt = /* @__PURE__ */ O(Oo, [["render", Ho]]), Uo = {}, Go = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 48 48",
  width: "21px",
  height: "21px",
  "fill-rule": "evenodd",
  "clip-rule": "evenodd"
}, Zo = /* @__PURE__ */ k("path", {
  fill: "#fff",
  "fill-rule": "evenodd",
  d: "M11.553,11.099c1.232,1.001,1.694,0.925,4.008,0.77 l21.812-1.31c0.463,0,0.078-0.461-0.076-0.538l-3.622-2.619c-0.694-0.539-1.619-1.156-3.391-1.002l-21.12,1.54 c-0.77,0.076-0.924,0.461-0.617,0.77L11.553,11.099z",
  "clip-rule": "evenodd"
}, null, -1), Yo = /* @__PURE__ */ k("path", {
  fill: "#fff",
  "fill-rule": "evenodd",
  d: "M12.862,16.182v22.95c0,1.233,0.616,1.695,2.004,1.619 l23.971-1.387c1.388-0.076,1.543-0.925,1.543-1.927V14.641c0-1-0.385-1.54-1.234-1.463l-25.05,1.463 C13.171,14.718,12.862,15.181,12.862,16.182L12.862,16.182z",
  "clip-rule": "evenodd"
}, null, -1), qo = /* @__PURE__ */ k("path", {
  fill: "#424242",
  "fill-rule": "evenodd",
  d: "M11.553,11.099c1.232,1.001,1.694,0.925,4.008,0.77 l21.812-1.31c0.463,0,0.078-0.461-0.076-0.538l-3.622-2.619c-0.694-0.539-1.619-1.156-3.391-1.002l-21.12,1.54 c-0.77,0.076-0.924,0.461-0.617,0.77L11.553,11.099z M12.862,16.182v22.95c0,1.233,0.616,1.695,2.004,1.619l23.971-1.387 c1.388-0.076,1.543-0.925,1.543-1.927V14.641c0-1-0.385-1.54-1.234-1.463l-25.05,1.463C13.171,14.718,12.862,15.181,12.862,16.182 L12.862,16.182z M36.526,17.413c0.154,0.694,0,1.387-0.695,1.465l-1.155,0.23v16.943c-1.003,0.539-1.928,0.847-2.698,0.847 c-1.234,0-1.543-0.385-2.467-1.54l-7.555-11.86v11.475l2.391,0.539c0,0,0,1.386-1.929,1.386l-5.317,0.308 c-0.154-0.308,0-1.078,0.539-1.232l1.388-0.385V20.418l-1.927-0.154c-0.155-0.694,0.23-1.694,1.31-1.772l5.704-0.385l7.862,12.015 V19.493l-2.005-0.23c-0.154-0.848,0.462-1.464,1.233-1.54L36.526,17.413z M7.389,5.862l21.968-1.618 c2.698-0.231,3.392-0.076,5.087,1.155l7.013,4.929C42.614,11.176,43,11.407,43,12.33v27.032c0,1.694-0.617,2.696-2.775,2.849 l-25.512,1.541c-1.62,0.077-2.391-0.154-3.239-1.232l-5.164-6.7C5.385,34.587,5,33.664,5,32.585V8.556 C5,7.171,5.617,6.015,7.389,5.862z",
  "clip-rule": "evenodd"
}, null, -1), Ko = [
  Zo,
  Yo,
  qo
];
function Jo(t, a) {
  return b(), B("svg", Go, Ko);
}
const Xo = /* @__PURE__ */ O(Uo, [["render", Jo]]), Qo = {}, ei = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 48 48",
  width: "21px",
  height: "21px"
}, ti = /* @__PURE__ */ Ne('<path fill="#33d375" d="M33,8c0-2.209-1.791-4-4-4s-4,1.791-4,4c0,1.254,0,9.741,0,11c0,2.209,1.791,4,4,4s4-1.791,4-4	C33,17.741,33,9.254,33,8z"></path><path fill="#33d375" d="M43,19c0,2.209-1.791,4-4,4c-1.195,0-4,0-4,0s0-2.986,0-4c0-2.209,1.791-4,4-4S43,16.791,43,19z"></path><path fill="#40c4ff" d="M8,14c-2.209,0-4,1.791-4,4s1.791,4,4,4c1.254,0,9.741,0,11,0c2.209,0,4-1.791,4-4s-1.791-4-4-4	C17.741,14,9.254,14,8,14z"></path><path fill="#40c4ff" d="M19,4c2.209,0,4,1.791,4,4c0,1.195,0,4,0,4s-2.986,0-4,0c-2.209,0-4-1.791-4-4S16.791,4,19,4z"></path><path fill="#e91e63" d="M14,39.006C14,41.212,15.791,43,18,43s4-1.788,4-3.994c0-1.252,0-9.727,0-10.984	c0-2.206-1.791-3.994-4-3.994s-4,1.788-4,3.994C14,29.279,14,37.754,14,39.006z"></path><path fill="#e91e63" d="M4,28.022c0-2.206,1.791-3.994,4-3.994c1.195,0,4,0,4,0s0,2.981,0,3.994c0,2.206-1.791,3.994-4,3.994	S4,30.228,4,28.022z"></path><path fill="#ffc107" d="M39,33c2.209,0,4-1.791,4-4s-1.791-4-4-4c-1.254,0-9.741,0-11,0c-2.209,0-4,1.791-4,4s1.791,4,4,4	C29.258,33,37.746,33,39,33z"></path><path fill="#ffc107" d="M28,43c-2.209,0-4-1.791-4-4c0-1.195,0-4,0-4s2.986,0,4,0c2.209,0,4,1.791,4,4S30.209,43,28,43z"></path>', 8), ai = [
  ti
];
function ni(t, a) {
  return b(), B("svg", ei, ai);
}
const ri = /* @__PURE__ */ O(Qo, [["render", ni]]), oi = {}, ii = {
  width: "21",
  height: "21",
  viewBox: "0 0 512 512",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
}, li = /* @__PURE__ */ k("path", {
  d: "M255.498 31.0034C131.513 31.0034 31 131.515 31 255.502C31 379.492 131.513 480 255.498 480C379.497 480 480 379.495 480 255.502C480 131.522 379.497 31.0135 255.495 31.0135L255.498 31V31.0034ZM358.453 354.798C354.432 361.391 345.801 363.486 339.204 359.435C286.496 327.237 220.139 319.947 141.993 337.801C134.463 339.516 126.957 334.798 125.24 327.264C123.516 319.731 128.217 312.225 135.767 310.511C221.284 290.972 294.639 299.384 353.816 335.549C360.413 339.596 362.504 348.2 358.453 354.798ZM385.932 293.67C380.864 301.903 370.088 304.503 361.858 299.438C301.512 262.345 209.528 251.602 138.151 273.272C128.893 276.067 119.118 270.851 116.309 261.61C113.521 252.353 118.74 242.597 127.981 239.782C209.512 215.044 310.87 227.026 380.17 269.612C388.4 274.68 391 285.456 385.935 293.676V293.673L385.932 293.67ZM388.293 230.016C315.935 187.039 196.56 183.089 127.479 204.055C116.387 207.42 104.654 201.159 101.293 190.063C97.9326 178.964 104.189 167.241 115.289 163.87C194.59 139.796 326.418 144.446 409.723 193.902C419.722 199.826 422.995 212.71 417.068 222.675C411.168 232.653 398.247 235.943 388.303 230.016H388.293V230.016Z",
  fill: "#1ED760"
}, null, -1), si = [
  li
];
function ci(t, a) {
  return b(), B("svg", ii, si);
}
const pi = /* @__PURE__ */ O(oi, [["render", ci]]), di = {}, ui = {
  width: "21",
  height: "21",
  viewBox: "0 0 512 512",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
}, fi = /* @__PURE__ */ k("path", {
  d: "M416 240L352 304H288L232 360V304H160V64H416V240Z",
  fill: "white"
}, null, -1), hi = /* @__PURE__ */ k("path", {
  d: "M144 32L64 112V400H160V480L240 400H304L448 256V32H144ZM416 240L352 304H288L232 360V304H160V64H416V240Z",
  fill: "#9146FF"
}, null, -1), gi = /* @__PURE__ */ k("path", {
  d: "M368 120H336V216H368V120Z",
  fill: "#9146FF"
}, null, -1), mi = /* @__PURE__ */ k("path", {
  d: "M280 120H248V216H280V120Z",
  fill: "#9146FF"
}, null, -1), _i = [
  fi,
  hi,
  gi,
  mi
];
function vi(t, a) {
  return b(), B("svg", ui, _i);
}
const bi = /* @__PURE__ */ O(di, [["render", vi]]), yi = {}, wi = {
  width: "21",
  height: "21",
  viewBox: "0 0 512 512",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
}, Ci = /* @__PURE__ */ k("path", {
  d: "M33 256.043C33 264.556 35.3159 273.069 39.4845 280.202L117.993 415.493C126.098 429.298 138.373 440.572 153.657 445.634C183.764 455.528 214.797 442.873 229.618 417.333L248.609 384.661L173.806 256.043L252.777 119.831L271.768 87.1591C277.557 77.2654 284.968 69.4424 294 63H285.894H172.185C150.878 63 131.193 74.2742 120.54 92.6812L39.7161 231.884C35.3159 239.016 33 247.53 33 256.043Z",
  fill: "#6363F1"
}, null, -1), Si = /* @__PURE__ */ k("path", {
  d: "M480 256.058C480 247.539 477.684 239.021 473.516 231.883L393.849 94.6596C379.028 69.3331 347.995 56.4396 317.888 66.34C302.603 71.4053 290.329 82.6871 282.224 96.5015L264.391 127.354L339.194 256.058L260.223 392.131L241.232 424.825C235.443 434.495 228.032 442.553 219 449H227.106H340.815C362.122 449 381.807 437.718 392.46 419.299L473.284 280.003C477.684 272.866 480 264.577 480 256.058Z",
  fill: "#6363F1"
}, null, -1), $i = [
  Ci,
  Si
];
function ki(t, a) {
  return b(), B("svg", wi, $i);
}
const Li = /* @__PURE__ */ O(yi, [["render", ki]]), Bi = {}, Ti = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 48 48",
  width: "21",
  height: "21"
}, xi = /* @__PURE__ */ k("circle", {
  cx: "24",
  cy: "24",
  r: "20",
  fill: "#2196f3"
}, null, -1), Mi = /* @__PURE__ */ k("path", {
  fill: "#fff",
  d: "M29,31H14c-1.657,0-3-1.343-3-3V17h15c1.657,0,3,1.343,3,3V31z"
}, null, -1), Ii = /* @__PURE__ */ k("polygon", {
  fill: "#fff",
  points: "37,31 31,27 31,21 37,17"
}, null, -1), zi = [
  xi,
  Mi,
  Ii
];
function Ai(t, a) {
  return b(), B("svg", Ti, zi);
}
const Pi = /* @__PURE__ */ O(Bi, [["render", Ai]]), Ei = /* @__PURE__ */ N({
  __name: "Icons",
  props: {
    name: {}
  },
  setup(t) {
    const a = {
      apple: Nr,
      azure: $o,
      bitbucket: uo,
      discord: vo,
      facebook: xr,
      figma: Zr,
      google: Cr,
      github: Qr,
      gitlab: oo,
      kakao: zo,
      keycloak: Ro,
      linkedin: qt,
      linkedin_oidc: qt,
      notion: Xo,
      slack: ri,
      spotify: pi,
      twitch: bi,
      twitter: Er,
      workos: Li,
      zoom: Pi
    };
    return (r, n) => (b(), L(be(a[r.name])));
  }
}), ji = /* @__PURE__ */ N({
  __name: "SocialAuth",
  props: {
    supabaseClient: {},
    socialLayout: { default: "vertical" },
    providers: { default: () => ["github", "google", "azure"] },
    providerScopes: {},
    queryParams: {},
    redirectTo: {},
    onlyThirdPartyProviders: { type: Boolean, default: !0 },
    view: { default: "sign_in" },
    i18n: {},
    appearance: {}
  },
  setup(t) {
    const a = t, r = z(""), n = z(!1), { authView: s } = ze(Se), l = R(() => a.socialLayout === "vertical"), h = R(() => s.value === "magic_link" ? "sign_in" : s.value), u = R(
      () => {
        var f;
        return (f = a.i18n) == null ? void 0 : f[h.value];
      }
    ), o = async (f) => {
      var p;
      r.value = "", n.value = !0;
      const { data: m, error: c } = await a.supabaseClient.auth.signInWithOAuth({
        provider: f,
        options: {
          redirectTo: a.redirectTo,
          scopes: (p = a.providerScopes) == null ? void 0 : p[f],
          queryParams: a.queryParams
        }
      });
      c && (r.value = c.message), n.value = !1;
    };
    function g(f) {
      const m = f.toLowerCase();
      return f.charAt(0).toUpperCase() + m.slice(1);
    }
    return (f, m) => f.providers && f.providers.length > 0 ? (b(), B(Fe, { key: 0 }, [
      I(_(re), {
        gap: "large",
        direction: "vertical",
        appearance: f.appearance
      }, {
        default: S(() => [
          I(_(re), {
            direction: l.value ? "vertical" : "horizontal",
            gap: l.value ? "small" : "medium",
            appearance: f.appearance
          }, {
            default: S(() => [
              (b(!0), B(Fe, null, kn(f.providers, (c) => (b(), L(_(Ie), {
                key: c,
                color: "default",
                loading: n.value,
                onClick: q(() => o(c), ["prevent"]),
                appearance: f.appearance
              }, {
                icon: S(() => [
                  I(_(Ei), { name: c }, null, 8, ["name"])
                ]),
                default: S(() => {
                  var p, v;
                  return [
                    l.value ? (b(), B(Fe, { key: 0 }, [
                      T(x(_(xn)(
                        (v = (p = u.value) == null ? void 0 : p.social_provider_text) == null ? void 0 : v.replace("_oidc", ""),
                        {
                          provider: g(c.replace("_oidc", ""))
                        }
                      )), 1)
                    ], 64)) : j("", !0)
                  ];
                }),
                _: 2
              }, 1032, ["loading", "onClick", "appearance"]))), 128))
            ]),
            _: 1
          }, 8, ["direction", "gap", "appearance"])
        ]),
        _: 1
      }, 8, ["appearance"]),
      f.onlyThirdPartyProviders ? j("", !0) : (b(), L(_(pr), {
        key: 0,
        appearance: f.appearance
      }, null, 8, ["appearance"]))
    ], 64)) : j("", !0);
  }
}), Vi = ["id", "onSubmit"], Ri = /* @__PURE__ */ N({
  __name: "MagicLink",
  props: {
    appearance: {},
    supabaseClient: {},
    redirectTo: {},
    showLinks: { type: Boolean },
    i18n: {}
  },
  setup(t) {
    const a = t, r = z(""), n = z(""), s = z(""), l = z(!1), { authView: h, setAuthView: u } = ze(Se), o = R(
      () => {
        var f;
        return (f = a.i18n) == null ? void 0 : f[h.value];
      }
    ), g = async (f) => {
      var c, p;
      n.value = "", s.value = "", l.value = !0;
      const { error: m } = await a.supabaseClient.auth.signInWithOtp({
        email: r.value,
        options: { emailRedirectTo: a.redirectTo }
      });
      m ? n.value = m.message : s.value = (p = (c = a.i18n) == null ? void 0 : c.magic_link) == null ? void 0 : p.confirmation_text, l.value = !1;
    };
    return (f, m) => (b(), B("form", {
      id: `auth-${_(h)}`,
      onSubmit: q(g, ["prevent"]),
      autoComplete: "on",
      style: {
        width: "100%"
      }
    }, [
      I(_(re), {
        direction: "vertical",
        gap: "large",
        appearance: f.appearance
      }, {
        default: S(() => {
          var c;
          return [
            k("div", null, [
              I(_(ce), {
                htmlFor: "email",
                appearance: f.appearance
              }, {
                default: S(() => {
                  var p;
                  return [
                    T(x((p = o.value) == null ? void 0 : p.email_input_label), 1)
                  ];
                }),
                _: 1
              }, 8, ["appearance"]),
              I(_(se), {
                id: "email",
                type: "email",
                name: "email",
                autoComplete: "email",
                autofocus: "",
                placeholder: (c = o.value) == null ? void 0 : c.email_input_placeholder,
                appearance: f.appearance,
                modelValue: r.value,
                "onUpdate:modelValue": m[0] || (m[0] = (p) => r.value = p)
              }, null, 8, ["placeholder", "appearance", "modelValue"])
            ]),
            I(_(Ie), {
              type: "submit",
              color: "primary",
              loading: l.value,
              appearance: f.appearance
            }, {
              default: S(() => {
                var p, v;
                return [
                  T(x(l.value ? (p = o.value) == null ? void 0 : p.loading_button_label : (v = o.value) == null ? void 0 : v.button_label), 1)
                ];
              }),
              _: 1
            }, 8, ["loading", "appearance"]),
            f.showLinks ? (b(), L(_(me), {
              key: 0,
              href: "#auth-sign-in",
              appearance: f.appearance,
              onClick: m[1] || (m[1] = q((p) => _(u)(_(G).SIGN_IN), ["prevent"]))
            }, {
              default: S(() => {
                var p, v;
                return [
                  T(x((v = (p = f.i18n) == null ? void 0 : p.sign_in) == null ? void 0 : v.link_text), 1)
                ];
              }),
              _: 1
            }, 8, ["appearance"])) : j("", !0)
          ];
        }),
        _: 1
      }, 8, ["appearance"]),
      s.value ? (b(), L(_(ee), {
        key: 0,
        appearance: f.appearance
      }, {
        default: S(() => [
          T(x(s.value), 1)
        ]),
        _: 1
      }, 8, ["appearance"])) : j("", !0),
      n.value ? (b(), L(_(ee), {
        key: 1,
        color: "danger",
        appearance: f.appearance
      }, {
        default: S(() => [
          T(x(n.value), 1)
        ]),
        _: 1
      }, 8, ["appearance"])) : j("", !0)
    ], 40, Vi));
  }
}), Oi = ["id", "onSubmit"], Fi = /* @__PURE__ */ N({
  __name: "ForgottenPassword",
  props: {
    appearance: {},
    supabaseClient: {},
    redirectTo: {},
    showLinks: { type: Boolean },
    i18n: {}
  },
  setup(t) {
    const a = t, r = z(""), n = z(""), s = z(""), l = z(!1), { authView: h, setAuthView: u } = ze(Se), o = R(
      () => {
        var f;
        return (f = a.i18n) == null ? void 0 : f[h.value];
      }
    ), g = async (f) => {
      var c, p;
      n.value = "", s.value = "", l.value = !0;
      const { error: m } = await a.supabaseClient.auth.resetPasswordForEmail(r.value, {
        redirectTo: a.redirectTo
      });
      m ? n.value = m.message : s.value = (p = (c = a.i18n) == null ? void 0 : c.forgotten_password) == null ? void 0 : p.confirmation_text, l.value = !1;
    };
    return (f, m) => (b(), B("form", {
      id: `auth-${_(h)}`,
      onSubmit: q(g, ["prevent"]),
      autoComplete: "on",
      style: {
        width: "100%"
      }
    }, [
      I(_(re), {
        direction: "vertical",
        gap: "large",
        appearance: f.appearance
      }, {
        default: S(() => {
          var c;
          return [
            k("div", null, [
              I(_(ce), {
                htmlFor: "email",
                appearance: f.appearance
              }, {
                default: S(() => {
                  var p;
                  return [
                    T(x((p = o.value) == null ? void 0 : p.email_label), 1)
                  ];
                }),
                _: 1
              }, 8, ["appearance"]),
              I(_(se), {
                id: "email",
                type: "email",
                name: "email",
                autoComplete: "email",
                autofocus: "",
                placeholder: (c = o.value) == null ? void 0 : c.email_input_placeholder,
                appearance: f.appearance,
                modelValue: r.value,
                "onUpdate:modelValue": m[0] || (m[0] = (p) => r.value = p)
              }, null, 8, ["placeholder", "appearance", "modelValue"])
            ]),
            I(_(Ie), {
              type: "submit",
              color: "primary",
              loading: l.value,
              appearance: f.appearance
            }, {
              default: S(() => {
                var p, v;
                return [
                  T(x(l.value ? (p = o.value) == null ? void 0 : p.loading_button_label : (v = o.value) == null ? void 0 : v.button_label), 1)
                ];
              }),
              _: 1
            }, 8, ["loading", "appearance"]),
            f.showLinks ? (b(), L(_(me), {
              key: 0,
              href: "#auth-sign-in",
              appearance: f.appearance,
              onClick: m[1] || (m[1] = q((p) => _(u)(_(G).SIGN_IN), ["prevent"]))
            }, {
              default: S(() => {
                var p, v;
                return [
                  T(x((v = (p = f.i18n) == null ? void 0 : p.sign_in) == null ? void 0 : v.link_text), 1)
                ];
              }),
              _: 1
            }, 8, ["appearance"])) : j("", !0)
          ];
        }),
        _: 1
      }, 8, ["appearance"]),
      s.value ? (b(), L(_(ee), {
        key: 0,
        appearance: f.appearance
      }, {
        default: S(() => [
          T(x(s.value), 1)
        ]),
        _: 1
      }, 8, ["appearance"])) : j("", !0),
      n.value ? (b(), L(_(ee), {
        key: 1,
        color: "danger",
        appearance: f.appearance
      }, {
        default: S(() => [
          T(x(n.value), 1)
        ]),
        _: 1
      }, 8, ["appearance"])) : j("", !0)
    ], 40, Oi));
  }
}), Ni = ["id", "onSubmit"], Wi = /* @__PURE__ */ N({
  __name: "UpdatePassword",
  props: {
    appearance: {},
    supabaseClient: {},
    i18n: {}
  },
  setup(t) {
    const a = t, r = z(""), n = z(""), s = z(""), l = z(!1), { authView: h } = ze(Se), u = R(
      () => {
        var g;
        return (g = a.i18n) == null ? void 0 : g[h.value];
      }
    ), o = async (g) => {
      var m, c;
      n.value = "", s.value = "", l.value = !0;
      const { error: f } = await a.supabaseClient.auth.updateUser({
        password: r.value
      });
      f ? n.value = f.message : s.value = (c = (m = a.i18n) == null ? void 0 : m.update_password) == null ? void 0 : c.confirmation_text, l.value = !1;
    };
    return (g, f) => (b(), B("form", {
      id: `auth-${_(h)}`,
      onSubmit: q(o, ["prevent"]),
      autoComplete: "on",
      style: {
        width: "100%"
      }
    }, [
      I(_(re), {
        direction: "vertical",
        gap: "large",
        appearance: g.appearance
      }, {
        default: S(() => {
          var m;
          return [
            k("div", null, [
              I(_(ce), {
                htmlFor: "password_label",
                appearance: g.appearance
              }, {
                default: S(() => {
                  var c;
                  return [
                    T(x((c = u.value) == null ? void 0 : c.password_label), 1)
                  ];
                }),
                _: 1
              }, 8, ["appearance"]),
              I(_(se), {
                id: "password",
                type: "password",
                name: "password",
                autofocus: "",
                placeholder: (m = u.value) == null ? void 0 : m.password_input_placeholder,
                appearance: g.appearance,
                modelValue: r.value,
                "onUpdate:modelValue": f[0] || (f[0] = (c) => r.value = c)
              }, null, 8, ["placeholder", "appearance", "modelValue"])
            ]),
            I(_(Ie), {
              type: "submit",
              color: "primary",
              loading: l.value,
              appearance: g.appearance
            }, {
              default: S(() => {
                var c, p;
                return [
                  T(x(l.value ? (c = u.value) == null ? void 0 : c.loading_button_label : (p = u.value) == null ? void 0 : p.button_label), 1)
                ];
              }),
              _: 1
            }, 8, ["loading", "appearance"])
          ];
        }),
        _: 1
      }, 8, ["appearance"]),
      s.value ? (b(), L(_(ee), {
        key: 0,
        appearance: g.appearance
      }, {
        default: S(() => [
          T(x(s.value), 1)
        ]),
        _: 1
      }, 8, ["appearance"])) : j("", !0),
      n.value ? (b(), L(_(ee), {
        key: 1,
        color: "danger",
        appearance: g.appearance
      }, {
        default: S(() => [
          T(x(n.value), 1)
        ]),
        _: 1
      }, 8, ["appearance"])) : j("", !0)
    ], 40, Ni));
  }
}), Di = ["id", "onSubmit"], Hi = { key: 0 }, Ui = { key: 1 }, Gi = /* @__PURE__ */ N({
  __name: "VerifyOtp",
  props: {
    appearance: {},
    supabaseClient: {},
    otpType: { default: "email" },
    showLinks: { type: Boolean, default: !1 },
    i18n: {}
  },
  setup(t) {
    const a = t, r = z(""), n = z(""), s = z(""), l = z(""), h = z(""), u = z(!1), { authView: o, setAuthView: g } = ze(Se), f = R(
      () => {
        var c;
        return (c = a.i18n) == null ? void 0 : c[o.value];
      }
    ), m = async (c) => {
      var $, P;
      l.value = "", h.value = "", u.value = !0;
      let p = {
        email: r.value,
        token: s.value,
        type: a.otpType
      };
      ["sms", "phone_change"].includes(a.otpType) && (p = {
        phone: n.value,
        token: s.value,
        type: a.otpType
      });
      const { error: v } = await a.supabaseClient.auth.verifyOtp(
        p
      );
      v ? l.value = v.message : h.value = (P = ($ = a.i18n) == null ? void 0 : $.magic_link) == null ? void 0 : P.confirmation_text, u.value = !1;
    };
    return (c, p) => (b(), B("form", {
      id: `auth-${_(o)}`,
      onSubmit: q(m, ["prevent"]),
      autoComplete: "on",
      style: {
        width: "100%"
      }
    }, [
      I(_(re), {
        direction: "vertical",
        gap: "large",
        appearance: c.appearance
      }, {
        default: S(() => {
          var v, $, P;
          return [
            ["sms", "phone_change"].includes(c.otpType) ? (b(), B("div", Hi, [
              I(_(ce), {
                htmlFor: "phone",
                appearance: c.appearance
              }, {
                default: S(() => {
                  var w;
                  return [
                    T(x((w = f.value) == null ? void 0 : w.phone_input_label), 1)
                  ];
                }),
                _: 1
              }, 8, ["appearance"]),
              I(_(se), {
                id: "phone",
                name: "phone",
                type: "text",
                autofocus: "",
                placeholder: (v = f.value) == null ? void 0 : v.phone_input_placeholder,
                appearance: c.appearance,
                modelValue: n.value,
                "onUpdate:modelValue": p[0] || (p[0] = (w) => n.value = w)
              }, null, 8, ["placeholder", "appearance", "modelValue"])
            ])) : (b(), B("div", Ui, [
              I(_(ce), {
                htmlFor: "email",
                appearance: c.appearance
              }, {
                default: S(() => {
                  var w;
                  return [
                    T(x((w = f.value) == null ? void 0 : w.email_input_label), 1)
                  ];
                }),
                _: 1
              }, 8, ["appearance"]),
              I(_(se), {
                id: "email",
                name: "email",
                type: "email",
                autoComplete: "email",
                autofocus: "",
                placeholder: ($ = f.value) == null ? void 0 : $.email_input_placeholder,
                appearance: c.appearance,
                modelValue: r.value,
                "onUpdate:modelValue": p[1] || (p[1] = (w) => r.value = w)
              }, null, 8, ["placeholder", "appearance", "modelValue"])
            ])),
            k("div", null, [
              I(_(ce), {
                htmlFor: "token",
                appearance: c.appearance
              }, {
                default: S(() => {
                  var w;
                  return [
                    T(x((w = f.value) == null ? void 0 : w.token_input_label), 1)
                  ];
                }),
                _: 1
              }, 8, ["appearance"]),
              I(_(se), {
                id: "token",
                name: "token",
                type: "text",
                autofocus: "",
                placeholder: (P = f.value) == null ? void 0 : P.token_input_placeholder,
                appearance: c.appearance,
                modelValue: s.value,
                "onUpdate:modelValue": p[2] || (p[2] = (w) => s.value = w)
              }, null, 8, ["placeholder", "appearance", "modelValue"])
            ]),
            I(_(Ie), {
              type: "submit",
              color: "primary",
              loading: u.value,
              appearance: c.appearance
            }, {
              default: S(() => {
                var w, M;
                return [
                  T(x(u.value ? (w = f.value) == null ? void 0 : w.loading_button_label : (M = f.value) == null ? void 0 : M.button_label), 1)
                ];
              }),
              _: 1
            }, 8, ["loading", "appearance"]),
            c.showLinks ? (b(), L(_(me), {
              key: 2,
              href: "#auth-sign-in",
              appearance: c.appearance,
              onClick: p[3] || (p[3] = q((w) => _(g)(_(G).SIGN_IN), ["prevent"]))
            }, {
              default: S(() => {
                var w, M;
                return [
                  T(x((M = (w = c.i18n) == null ? void 0 : w.sign_in) == null ? void 0 : M.link_text), 1)
                ];
              }),
              _: 1
            }, 8, ["appearance"])) : j("", !0)
          ];
        }),
        _: 1
      }, 8, ["appearance"]),
      h.value ? (b(), L(_(ee), {
        key: 0,
        appearance: c.appearance
      }, {
        default: S(() => [
          T(x(h.value), 1)
        ]),
        _: 1
      }, 8, ["appearance"])) : j("", !0),
      l.value ? (b(), L(_(ee), {
        key: 1,
        color: "danger",
        appearance: c.appearance
      }, {
        default: S(() => [
          T(x(l.value), 1)
        ]),
        _: 1
      }, 8, ["appearance"])) : j("", !0)
    ], 40, Di));
  }
}), Zi = /* @__PURE__ */ N({
  __name: "Auth",
  props: {
    appearance: {},
    supabaseClient: {},
    socialLayout: { default: "vertical" },
    providers: {},
    providerScopes: {},
    queryParams: {},
    view: { default: "sign_in" },
    redirectTo: {},
    onlyThirdPartyProviders: { type: Boolean, default: !1 },
    magicLink: { type: Boolean, default: !1 },
    showLinks: { type: Boolean, default: !0 },
    otpType: { default: "email" },
    additionalData: {},
    dark: { type: Boolean, default: !1 },
    localization: {},
    theme: { default: "default" }
  },
  emits: ["update:view"],
  setup(t, { emit: a }) {
    const r = t, n = z(r.view), s = (o) => {
      a("update:view", o), n.value = o;
    };
    Kt(Se, {
      authView: n,
      setAuthView: s
    });
    const l = R(() => {
      var f;
      const o = Ke(Mn), g = Ke((f = r == null ? void 0 : r.localization) == null ? void 0 : f.variables);
      return ht(o, g ?? {});
    }), h = R(() => n.value === "sign_in" || n.value === "sign_up" || n.value === "magic_link"), u = R(() => {
      var f, m;
      const o = Ke((f = r.appearance) == null ? void 0 : f.theme), g = Ke((m = r.appearance) == null ? void 0 : m.variables);
      return ht(
        (o == null ? void 0 : o.default) ?? {},
        (o == null ? void 0 : o[r.theme]) ?? {},
        (g == null ? void 0 : g.default) ?? {},
        (g == null ? void 0 : g[r == null ? void 0 : r.theme]) ?? {}
      );
    });
    return mt({ theme: u.value }), ut(
      () => [r.appearance, r.theme],
      () => {
        mt({ theme: u.value });
      },
      { deep: !0 }
    ), ut(
      () => r.view,
      (o) => {
        const { data: g } = r.supabaseClient.auth.onAuthStateChange(
          (f) => {
            f === "PASSWORD_RECOVERY" ? s("update_password") : f === "USER_UPDATED" && s("sign_in");
          }
        );
        return s(o), () => g.subscription.unsubscribe();
      },
      { deep: !0 }
    ), (o, g) => h.value ? (b(), L(sr, {
      key: 0,
      direction: "horizontal",
      gap: "small",
      theme: r.theme,
      appearance: o.appearance
    }, {
      default: S(() => [
        h.value ? (b(), L(ji, {
          key: 0,
          supabaseClient: o.supabaseClient,
          appearance: o.appearance,
          providers: o.providers,
          providerScopes: o.providerScopes,
          queryParams: o.queryParams,
          socialLayout: o.socialLayout,
          redirectTo: o.redirectTo,
          onlyThirdPartyProviders: o.onlyThirdPartyProviders,
          i18n: l.value
        }, null, 8, ["supabaseClient", "appearance", "providers", "providerScopes", "queryParams", "socialLayout", "redirectTo", "onlyThirdPartyProviders", "i18n"])) : j("", !0),
        o.onlyThirdPartyProviders ? j("", !0) : (b(), B(Fe, { key: 1 }, [
          n.value === "sign_in" || n.value === "sign_up" ? (b(), L(fr, {
            key: 0,
            supabaseClient: o.supabaseClient,
            appearance: o.appearance,
            redirectTo: o.redirectTo,
            magicLink: o.magicLink,
            showLinks: o.showLinks,
            i18n: l.value,
            additionalData: o.additionalData
          }, null, 8, ["supabaseClient", "appearance", "redirectTo", "magicLink", "showLinks", "i18n", "additionalData"])) : j("", !0),
          n.value === "magic_link" ? (b(), L(Ri, {
            key: 1,
            supabaseClient: o.supabaseClient,
            appearance: o.appearance,
            redirectTo: o.redirectTo,
            showLinks: o.showLinks,
            i18n: l.value
          }, null, 8, ["supabaseClient", "appearance", "redirectTo", "showLinks", "i18n"])) : j("", !0)
        ], 64))
      ]),
      _: 1
    }, 8, ["theme", "appearance"])) : (b(), B(Fe, { key: 1 }, [
      n.value === "forgotten_password" ? (b(), L(Fi, {
        key: 0,
        supabaseClient: o.supabaseClient,
        appearance: o.appearance,
        redirectTo: o.redirectTo,
        showLinks: o.showLinks,
        i18n: l.value
      }, null, 8, ["supabaseClient", "appearance", "redirectTo", "showLinks", "i18n"])) : j("", !0),
      n.value === "update_password" ? (b(), L(Wi, {
        key: 1,
        supabaseClient: o.supabaseClient,
        appearance: o.appearance,
        i18n: l.value
      }, null, 8, ["supabaseClient", "appearance", "i18n"])) : j("", !0),
      n.value === "verify_otp" ? (b(), L(Gi, {
        key: 2,
        supabaseClient: o.supabaseClient,
        appearance: o.appearance,
        otpType: o.otpType,
        i18n: l.value
      }, null, 8, ["supabaseClient", "appearance", "otpType", "i18n"])) : j("", !0)
    ], 64));
  }
}), qi = N({
  name: "UserContextProvider",
  props: {
    supabaseClient: {
      type: Object,
      required: !0
    }
  },
  setup({ supabaseClient: t }, { slots: a }) {
    const r = z(null), n = z(null);
    return Jt(async () => {
      var h;
      const { data: s } = await t.auth.getSession();
      r.value = s.session, n.value = ((h = s.session) == null ? void 0 : h.user) ?? null;
      const { data: l } = t.auth.onAuthStateChange(
        async (u, o) => {
          r.value = o, n.value = (o == null ? void 0 : o.user) ?? null;
        }
      );
      return () => {
        l == null || l.subscription.unsubscribe();
      };
    }), Kt(lr, {
      session: r,
      user: n
    }), () => {
      var s;
      return Ln("div", null, (s = a == null ? void 0 : a.default) == null ? void 0 : s.call(a));
    };
  }
}), Ki = (t) => {
  const a = z(null), r = (n) => {
    n ? JSON.stringify(a.value) !== JSON.stringify(n.user) && (a.value = n.user) : a.value = null;
  };
  return t == null || t.auth.getSession().then(({ data: { session: n } }) => {
    n && r(n);
  }), Jt(async () => {
    const { data: n } = t.auth.onAuthStateChange(
      async (s, l) => {
        l && r(l);
      }
    );
    return () => {
      n == null || n.subscription.unsubscribe();
    };
  }), {
    supabaseUser: a
  };
}, Ji = {
  install(t) {
    t.component("Auth", Zi);
  }
};
export {
  Zi as Auth,
  Fi as ForgottenPassword,
  Ri as MagicLink,
  Wi as UpdatePassword,
  qi as UserContextProvider,
  lr as UserContextProviderKey,
  Ji as default,
  ze as injectStrict,
  Ki as useSupabaseUser
};
